<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 13 Clustering | Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor</title>
  <meta name="description" content="Online textbook on ‘Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 13 Clustering | Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Online textbook on ‘Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor’" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 13 Clustering | Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor" />
  
  <meta name="twitter:description" content="Online textbook on ‘Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor’" />
  



<meta name="date" content="2021-03-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="dimensionality-reduction.html"/>
<link rel="next" href="marker-genes.html"/>
<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">OSTA</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#contents-of-this-book"><i class="fa fa-check"></i><b>1.1</b> Contents of this book</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#who-this-book-is-for"><i class="fa fa-check"></i><b>1.2</b> Who this book is for</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#bioconductor"><i class="fa fa-check"></i><b>1.3</b> Bioconductor</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="spatially-resolved-transcriptomics.html"><a href="spatially-resolved-transcriptomics.html"><i class="fa fa-check"></i><b>2</b> Spatially resolved transcriptomics</a>
<ul>
<li class="chapter" data-level="2.1" data-path="spatially-resolved-transcriptomics.html"><a href="spatially-resolved-transcriptomics.html#x-genomics-visium"><i class="fa fa-check"></i><b>2.1</b> 10x Genomics Visium</a></li>
<li class="chapter" data-level="2.2" data-path="spatially-resolved-transcriptomics.html"><a href="spatially-resolved-transcriptomics.html#slide-seqv2"><i class="fa fa-check"></i><b>2.2</b> Slide-seqV2</a></li>
<li class="chapter" data-level="2.3" data-path="spatially-resolved-transcriptomics.html"><a href="spatially-resolved-transcriptomics.html#seqfish"><i class="fa fa-check"></i><b>2.3</b> seqFISH</a></li>
<li class="chapter" data-level="2.4" data-path="spatially-resolved-transcriptomics.html"><a href="spatially-resolved-transcriptomics.html#merfish"><i class="fa fa-check"></i><b>2.4</b> MERFISH</a></li>
<li class="chapter" data-level="2.5" data-path="spatially-resolved-transcriptomics.html"><a href="spatially-resolved-transcriptomics.html#other-platforms"><i class="fa fa-check"></i><b>2.5</b> Other platforms</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spatialexperiment.html"><a href="spatialexperiment.html"><i class="fa fa-check"></i><b>3</b> SpatialExperiment</a>
<ul>
<li class="chapter" data-level="3.1" data-path="spatialexperiment.html"><a href="spatialexperiment.html#spatialexperiment-class"><i class="fa fa-check"></i><b>3.1</b> SpatialExperiment class</a></li>
</ul></li>
<li class="part"><span><b>II Preprocessing steps</b></span></li>
<li class="chapter" data-level="4" data-path="preprocessing-steps.html"><a href="preprocessing-steps.html"><i class="fa fa-check"></i><b>4</b> Preprocessing steps</a></li>
<li class="chapter" data-level="5" data-path="loupe-browser.html"><a href="loupe-browser.html"><i class="fa fa-check"></i><b>5</b> Loupe Browser</a>
<ul>
<li class="chapter" data-level="5.1" data-path="loupe-browser.html"><a href="loupe-browser.html#overview"><i class="fa fa-check"></i><b>5.1</b> Overview</a></li>
<li class="chapter" data-level="5.2" data-path="loupe-browser.html"><a href="loupe-browser.html#automatic-alignment-of-images"><i class="fa fa-check"></i><b>5.2</b> Automatic alignment of images</a></li>
<li class="chapter" data-level="5.3" data-path="loupe-browser.html"><a href="loupe-browser.html#manual-alignment-of-images"><i class="fa fa-check"></i><b>5.3</b> Manual alignment of images</a></li>
<li class="chapter" data-level="5.4" data-path="loupe-browser.html"><a href="loupe-browser.html#output-files-for-space-ranger"><i class="fa fa-check"></i><b>5.4</b> Output files for Space Ranger</a></li>
<li class="chapter" data-level="5.5" data-path="loupe-browser.html"><a href="loupe-browser.html#downstream-analyses-in-loupe-browser"><i class="fa fa-check"></i><b>5.5</b> Downstream analyses in Loupe Browser</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="image-segmentation.html"><a href="image-segmentation.html"><i class="fa fa-check"></i><b>6</b> Image segmentation</a>
<ul>
<li class="chapter" data-level="6.1" data-path="image-segmentation.html"><a href="image-segmentation.html#overview-1"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="image-segmentation.html"><a href="image-segmentation.html#split-histology-images"><i class="fa fa-check"></i><b>6.2</b> Split histology images</a></li>
<li class="chapter" data-level="6.3" data-path="image-segmentation.html"><a href="image-segmentation.html#file-size-issues"><i class="fa fa-check"></i><b>6.3</b> File size issues</a></li>
<li class="chapter" data-level="6.4" data-path="image-segmentation.html"><a href="image-segmentation.html#segment-cell-nuclei"><i class="fa fa-check"></i><b>6.4</b> Segment cell nuclei</a></li>
<li class="chapter" data-level="6.5" data-path="image-segmentation.html"><a href="image-segmentation.html#identify-number-of-cells-per-spot"><i class="fa fa-check"></i><b>6.5</b> Identify number of cells per spot</a></li>
<li class="chapter" data-level="6.6" data-path="image-segmentation.html"><a href="image-segmentation.html#output-files"><i class="fa fa-check"></i><b>6.6</b> Output files</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="space-ranger.html"><a href="space-ranger.html"><i class="fa fa-check"></i><b>7</b> Space Ranger</a>
<ul>
<li class="chapter" data-level="7.1" data-path="space-ranger.html"><a href="space-ranger.html#overview-2"><i class="fa fa-check"></i><b>7.1</b> Overview</a></li>
<li class="chapter" data-level="7.2" data-path="space-ranger.html"><a href="space-ranger.html#installation"><i class="fa fa-check"></i><b>7.2</b> Installation</a></li>
<li class="chapter" data-level="7.3" data-path="space-ranger.html"><a href="space-ranger.html#memory-issues"><i class="fa fa-check"></i><b>7.3</b> Memory issues</a></li>
<li class="chapter" data-level="7.4" data-path="space-ranger.html"><a href="space-ranger.html#run-spaceranger-count"><i class="fa fa-check"></i><b>7.4</b> Run spaceranger count</a></li>
<li class="chapter" data-level="7.5" data-path="space-ranger.html"><a href="space-ranger.html#output-files-1"><i class="fa fa-check"></i><b>7.5</b> Output files</a></li>
<li class="chapter" data-level="7.6" data-path="space-ranger.html"><a href="space-ranger.html#web-summary-.html-file"><i class="fa fa-check"></i><b>7.6</b> Web summary .html file</a></li>
<li class="chapter" data-level="7.7" data-path="space-ranger.html"><a href="space-ranger.html#import-outputs-into-r"><i class="fa fa-check"></i><b>7.7</b> Import outputs into R</a></li>
</ul></li>
<li class="part"><span><b>III Analysis steps</b></span></li>
<li class="chapter" data-level="8" data-path="analysis-steps.html"><a href="analysis-steps.html"><i class="fa fa-check"></i><b>8</b> Analysis steps</a>
<ul>
<li class="chapter" data-level="8.1" data-path="analysis-steps.html"><a href="analysis-steps.html#load-data"><i class="fa fa-check"></i><b>8.1</b> Load data</a></li>
<li class="chapter" data-level="8.2" data-path="analysis-steps.html"><a href="analysis-steps.html#spatialexperiment-object-structure"><i class="fa fa-check"></i><b>8.2</b> SpatialExperiment object structure</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>9</b> Quality control</a>
<ul>
<li class="chapter" data-level="9.1" data-path="quality-control.html"><a href="quality-control.html#background"><i class="fa fa-check"></i><b>9.1</b> Background</a></li>
<li class="chapter" data-level="9.2" data-path="quality-control.html"><a href="quality-control.html#load-data-1"><i class="fa fa-check"></i><b>9.2</b> Load data</a></li>
<li class="chapter" data-level="9.3" data-path="quality-control.html"><a href="quality-control.html#plot-data"><i class="fa fa-check"></i><b>9.3</b> Plot data</a></li>
<li class="chapter" data-level="9.4" data-path="quality-control.html"><a href="quality-control.html#calculate-qc-metrics"><i class="fa fa-check"></i><b>9.4</b> Calculate QC metrics</a></li>
<li class="chapter" data-level="9.5" data-path="quality-control.html"><a href="quality-control.html#selecting-thresholds"><i class="fa fa-check"></i><b>9.5</b> Selecting thresholds</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="quality-control.html"><a href="quality-control.html#library-size"><i class="fa fa-check"></i><b>9.5.1</b> Library size</a></li>
<li class="chapter" data-level="9.5.2" data-path="quality-control.html"><a href="quality-control.html#number-of-expressed-features"><i class="fa fa-check"></i><b>9.5.2</b> Number of expressed features</a></li>
<li class="chapter" data-level="9.5.3" data-path="quality-control.html"><a href="quality-control.html#proportion-of-mitochondrial-reads"><i class="fa fa-check"></i><b>9.5.3</b> Proportion of mitochondrial reads</a></li>
<li class="chapter" data-level="9.5.4" data-path="quality-control.html"><a href="quality-control.html#number-of-cells-per-spot"><i class="fa fa-check"></i><b>9.5.4</b> Number of cells per spot</a></li>
<li class="chapter" data-level="9.5.5" data-path="quality-control.html"><a href="quality-control.html#remove-low-quality-spots"><i class="fa fa-check"></i><b>9.5.5</b> Remove low-quality spots</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="quality-control.html"><a href="quality-control.html#zero-cell-and-single-cell-spots"><i class="fa fa-check"></i><b>9.6</b> Zero-cell and single-cell spots</a></li>
<li class="chapter" data-level="9.7" data-path="quality-control.html"><a href="quality-control.html#quality-control-at-gene-level"><i class="fa fa-check"></i><b>9.7</b> Quality control at gene level</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="normalization.html"><a href="normalization.html"><i class="fa fa-check"></i><b>10</b> Normalization</a>
<ul>
<li class="chapter" data-level="10.1" data-path="normalization.html"><a href="normalization.html#background-1"><i class="fa fa-check"></i><b>10.1</b> Background</a></li>
<li class="chapter" data-level="10.2" data-path="normalization.html"><a href="normalization.html#previous-steps"><i class="fa fa-check"></i><b>10.2</b> Previous steps</a></li>
<li class="chapter" data-level="10.3" data-path="normalization.html"><a href="normalization.html#log-transformed-normalized-counts"><i class="fa fa-check"></i><b>10.3</b> Log-transformed normalized counts</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="feature-selection.html"><a href="feature-selection.html"><i class="fa fa-check"></i><b>11</b> Feature selection</a>
<ul>
<li class="chapter" data-level="11.1" data-path="feature-selection.html"><a href="feature-selection.html#background-2"><i class="fa fa-check"></i><b>11.1</b> Background</a></li>
<li class="chapter" data-level="11.2" data-path="feature-selection.html"><a href="feature-selection.html#previous-steps-1"><i class="fa fa-check"></i><b>11.2</b> Previous steps</a></li>
<li class="chapter" data-level="11.3" data-path="feature-selection.html"><a href="feature-selection.html#highly-variable-genes-hvgs"><i class="fa fa-check"></i><b>11.3</b> Highly variable genes (HVGs)</a></li>
<li class="chapter" data-level="11.4" data-path="feature-selection.html"><a href="feature-selection.html#spatially-variable-genes-svgs"><i class="fa fa-check"></i><b>11.4</b> Spatially variable genes (SVGs)</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="feature-selection.html"><a href="feature-selection.html#spatialde"><i class="fa fa-check"></i><b>11.4.1</b> SpatialDE</a></li>
<li class="chapter" data-level="11.4.2" data-path="feature-selection.html"><a href="feature-selection.html#spark"><i class="fa fa-check"></i><b>11.4.2</b> SPARK</a></li>
<li class="chapter" data-level="11.4.3" data-path="feature-selection.html"><a href="feature-selection.html#morans-i-statistic"><i class="fa fa-check"></i><b>11.4.3</b> Moran’s I statistic</a></li>
<li class="chapter" data-level="11.4.4" data-path="feature-selection.html"><a href="feature-selection.html#further-methods"><i class="fa fa-check"></i><b>11.4.4</b> Further methods</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="dimensionality-reduction.html"><a href="dimensionality-reduction.html"><i class="fa fa-check"></i><b>12</b> Dimensionality reduction</a>
<ul>
<li class="chapter" data-level="12.1" data-path="dimensionality-reduction.html"><a href="dimensionality-reduction.html#background-3"><i class="fa fa-check"></i><b>12.1</b> Background</a></li>
<li class="chapter" data-level="12.2" data-path="dimensionality-reduction.html"><a href="dimensionality-reduction.html#previous-steps-2"><i class="fa fa-check"></i><b>12.2</b> Previous steps</a></li>
<li class="chapter" data-level="12.3" data-path="dimensionality-reduction.html"><a href="dimensionality-reduction.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>12.3</b> Principal component analysis (PCA)</a></li>
<li class="chapter" data-level="12.4" data-path="dimensionality-reduction.html"><a href="dimensionality-reduction.html#uniform-manifold-approximation-and-projection-umap"><i class="fa fa-check"></i><b>12.4</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="12.5" data-path="dimensionality-reduction.html"><a href="dimensionality-reduction.html#visualizations"><i class="fa fa-check"></i><b>12.5</b> Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="clustering.html"><a href="clustering.html"><i class="fa fa-check"></i><b>13</b> Clustering</a>
<ul>
<li class="chapter" data-level="13.1" data-path="clustering.html"><a href="clustering.html#overview-3"><i class="fa fa-check"></i><b>13.1</b> Overview</a></li>
<li class="chapter" data-level="13.2" data-path="clustering.html"><a href="clustering.html#previous-steps-3"><i class="fa fa-check"></i><b>13.2</b> Previous steps</a></li>
<li class="chapter" data-level="13.3" data-path="clustering.html"><a href="clustering.html#clustering-on-hvgs"><i class="fa fa-check"></i><b>13.3</b> Clustering on HVGs</a></li>
<li class="chapter" data-level="13.4" data-path="clustering.html"><a href="clustering.html#clustering-on-hvgs-and-spatial-dimensions"><i class="fa fa-check"></i><b>13.4</b> Clustering on HVGs and spatial dimensions</a></li>
<li class="chapter" data-level="13.5" data-path="clustering.html"><a href="clustering.html#bayesspace"><i class="fa fa-check"></i><b>13.5</b> BayesSpace</a></li>
<li class="chapter" data-level="13.6" data-path="clustering.html"><a href="clustering.html#further-methods-1"><i class="fa fa-check"></i><b>13.6</b> Further methods</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="marker-genes.html"><a href="marker-genes.html"><i class="fa fa-check"></i><b>14</b> Marker genes</a>
<ul>
<li class="chapter" data-level="14.1" data-path="marker-genes.html"><a href="marker-genes.html#background-4"><i class="fa fa-check"></i><b>14.1</b> Background</a></li>
<li class="chapter" data-level="14.2" data-path="marker-genes.html"><a href="marker-genes.html#previous-steps-4"><i class="fa fa-check"></i><b>14.2</b> Previous steps</a></li>
<li class="chapter" data-level="14.3" data-path="marker-genes.html"><a href="marker-genes.html#marker-genes-1"><i class="fa fa-check"></i><b>14.3</b> Marker genes</a></li>
</ul></li>
<li class="part"><span><b>IV Workflows</b></span></li>
<li class="chapter" data-level="15" data-path="workflows.html"><a href="workflows.html"><i class="fa fa-check"></i><b>15</b> Workflows</a></li>
<li class="chapter" data-level="16" data-path="human-dlpfc-workflow.html"><a href="human-dlpfc-workflow.html"><i class="fa fa-check"></i><b>16</b> Human DLPFC workflow</a>
<ul>
<li class="chapter" data-level="16.1" data-path="human-dlpfc-workflow.html"><a href="human-dlpfc-workflow.html#description-of-dataset"><i class="fa fa-check"></i><b>16.1</b> Description of dataset</a></li>
<li class="chapter" data-level="16.2" data-path="human-dlpfc-workflow.html"><a href="human-dlpfc-workflow.html#load-data-2"><i class="fa fa-check"></i><b>16.2</b> Load data</a></li>
<li class="chapter" data-level="16.3" data-path="human-dlpfc-workflow.html"><a href="human-dlpfc-workflow.html#plot-data-1"><i class="fa fa-check"></i><b>16.3</b> Plot data</a></li>
<li class="chapter" data-level="16.4" data-path="human-dlpfc-workflow.html"><a href="human-dlpfc-workflow.html#quality-control-qc"><i class="fa fa-check"></i><b>16.4</b> Quality control (QC)</a></li>
<li class="chapter" data-level="16.5" data-path="human-dlpfc-workflow.html"><a href="human-dlpfc-workflow.html#normalization-1"><i class="fa fa-check"></i><b>16.5</b> Normalization</a></li>
<li class="chapter" data-level="16.6" data-path="human-dlpfc-workflow.html"><a href="human-dlpfc-workflow.html#feature-selection-1"><i class="fa fa-check"></i><b>16.6</b> Feature selection</a></li>
<li class="chapter" data-level="16.7" data-path="human-dlpfc-workflow.html"><a href="human-dlpfc-workflow.html#dimensionality-reduction-1"><i class="fa fa-check"></i><b>16.7</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="16.8" data-path="human-dlpfc-workflow.html"><a href="human-dlpfc-workflow.html#clustering-1"><i class="fa fa-check"></i><b>16.8</b> Clustering</a></li>
<li class="chapter" data-level="16.9" data-path="human-dlpfc-workflow.html"><a href="human-dlpfc-workflow.html#marker-genes-2"><i class="fa fa-check"></i><b>16.9</b> Marker genes</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html"><i class="fa fa-check"></i><b>17</b> Mouse coronal workflow</a>
<ul>
<li class="chapter" data-level="17.1" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html#description-of-dataset-1"><i class="fa fa-check"></i><b>17.1</b> Description of dataset</a></li>
<li class="chapter" data-level="17.2" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html#load-data-3"><i class="fa fa-check"></i><b>17.2</b> Load data</a></li>
<li class="chapter" data-level="17.3" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html#plot-data-2"><i class="fa fa-check"></i><b>17.3</b> Plot data</a></li>
<li class="chapter" data-level="17.4" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html#quality-control-qc-1"><i class="fa fa-check"></i><b>17.4</b> Quality control (QC)</a></li>
<li class="chapter" data-level="17.5" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html#normalization-2"><i class="fa fa-check"></i><b>17.5</b> Normalization</a></li>
<li class="chapter" data-level="17.6" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html#feature-selection-2"><i class="fa fa-check"></i><b>17.6</b> Feature selection</a></li>
<li class="chapter" data-level="17.7" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html#dimensionality-reduction-2"><i class="fa fa-check"></i><b>17.7</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="17.8" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html#clustering-2"><i class="fa fa-check"></i><b>17.8</b> Clustering</a></li>
<li class="chapter" data-level="17.9" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html#marker-genes-3"><i class="fa fa-check"></i><b>17.9</b> Marker genes</a></li>
</ul></li>
<li class="part"><span><b>V Appendix</b></span></li>
<li class="chapter" data-level="18" data-path="contributors.html"><a href="contributors.html"><i class="fa fa-check"></i><b>18</b> Contributors</a></li>
<li class="chapter" data-level="19" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>19</b> References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="clustering" class="section level1" number="13">
<h1><span class="header-section-number">Chapter 13</span> Clustering</h1>
<div id="overview-3" class="section level2" number="13.1">
<h2><span class="header-section-number">13.1</span> Overview</h2>
<p>Clustering is used in single-cell and spatial analysis to identify cell types.</p>
<p>The definition of “cell type” is somewhat nebulous, and can depend to some extent on the biological context. Major cell types are generally well-defined and well-characterized (e.g. T cells, B cells, etc), while the definitions of more subtle or higher-resolution cell types may be somewhat less clear.</p>
<p>In the spatial context, we may be interested in e.g. (i) identifying cell types that occur in biologically interesting spatial patterns, or (ii) identifying major cell types and performing subsequent analyses within these cell types.</p>
</div>
<div id="previous-steps-3" class="section level2" number="13.2">
<h2><span class="header-section-number">13.2</span> Previous steps</h2>
<p><em>Code to run steps from the previous chapters, to generate the <code>SpatialExperiment</code> object required for this chapter.</em></p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r fold-hide"><code class="sourceCode r"><span id="cb101-1"><a href="clustering.html#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ---------</span></span>
<span id="cb101-2"><a href="clustering.html#cb101-2" aria-hidden="true" tabindex="-1"></a><span class="co"># load data</span></span>
<span id="cb101-3"><a href="clustering.html#cb101-3" aria-hidden="true" tabindex="-1"></a><span class="co"># ---------</span></span>
<span id="cb101-4"><a href="clustering.html#cb101-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-5"><a href="clustering.html#cb101-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SpatialExperiment)</span>
<span id="cb101-6"><a href="clustering.html#cb101-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(STexampleData)</span>
<span id="cb101-7"><a href="clustering.html#cb101-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-8"><a href="clustering.html#cb101-8" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> <span class="fu">load_data</span>(<span class="st">&quot;Visium_humanDLPFC&quot;</span>)</span>
<span id="cb101-9"><a href="clustering.html#cb101-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-10"><a href="clustering.html#cb101-10" aria-hidden="true" tabindex="-1"></a><span class="co"># --------------------</span></span>
<span id="cb101-11"><a href="clustering.html#cb101-11" aria-hidden="true" tabindex="-1"></a><span class="co"># quality control (QC)</span></span>
<span id="cb101-12"><a href="clustering.html#cb101-12" aria-hidden="true" tabindex="-1"></a><span class="co"># --------------------</span></span>
<span id="cb101-13"><a href="clustering.html#cb101-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-14"><a href="clustering.html#cb101-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb101-15"><a href="clustering.html#cb101-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-16"><a href="clustering.html#cb101-16" aria-hidden="true" tabindex="-1"></a><span class="co"># subset to keep only spots over tissue</span></span>
<span id="cb101-17"><a href="clustering.html#cb101-17" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> spe[, <span class="fu">spatialData</span>(spe)<span class="sc">$</span>in_tissue <span class="sc">==</span> <span class="dv">1</span>]</span>
<span id="cb101-18"><a href="clustering.html#cb101-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-19"><a href="clustering.html#cb101-19" aria-hidden="true" tabindex="-1"></a><span class="co"># identify mitochondrial genes</span></span>
<span id="cb101-20"><a href="clustering.html#cb101-20" aria-hidden="true" tabindex="-1"></a>is_mito <span class="ot">&lt;-</span> <span class="fu">grepl</span>(<span class="st">&quot;(^MT-)|(^mt-)&quot;</span>, <span class="fu">rowData</span>(spe)<span class="sc">$</span>gene_name)</span>
<span id="cb101-21"><a href="clustering.html#cb101-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-22"><a href="clustering.html#cb101-22" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate per-spot QC metrics and store in colData</span></span>
<span id="cb101-23"><a href="clustering.html#cb101-23" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> <span class="fu">addPerCellQC</span>(spe, <span class="at">subsets =</span> <span class="fu">list</span>(<span class="at">mito =</span> is_mito))</span>
<span id="cb101-24"><a href="clustering.html#cb101-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-25"><a href="clustering.html#cb101-25" aria-hidden="true" tabindex="-1"></a><span class="co"># select QC thresholds</span></span>
<span id="cb101-26"><a href="clustering.html#cb101-26" aria-hidden="true" tabindex="-1"></a>qc_lib_size <span class="ot">&lt;-</span> <span class="fu">colData</span>(spe)<span class="sc">$</span>sum <span class="sc">&lt;</span> <span class="dv">500</span></span>
<span id="cb101-27"><a href="clustering.html#cb101-27" aria-hidden="true" tabindex="-1"></a>qc_detected <span class="ot">&lt;-</span> <span class="fu">colData</span>(spe)<span class="sc">$</span>detected <span class="sc">&lt;</span> <span class="dv">250</span></span>
<span id="cb101-28"><a href="clustering.html#cb101-28" aria-hidden="true" tabindex="-1"></a>qc_mito <span class="ot">&lt;-</span> <span class="fu">colData</span>(spe)<span class="sc">$</span>subsets_mito_percent <span class="sc">&gt;</span> <span class="dv">30</span></span>
<span id="cb101-29"><a href="clustering.html#cb101-29" aria-hidden="true" tabindex="-1"></a>qc_cell_count <span class="ot">&lt;-</span> <span class="fu">colData</span>(spe)<span class="sc">$</span>cell_count <span class="sc">&gt;</span> <span class="dv">12</span></span>
<span id="cb101-30"><a href="clustering.html#cb101-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-31"><a href="clustering.html#cb101-31" aria-hidden="true" tabindex="-1"></a><span class="co"># combined set of discarded spots</span></span>
<span id="cb101-32"><a href="clustering.html#cb101-32" aria-hidden="true" tabindex="-1"></a>discard <span class="ot">&lt;-</span> qc_lib_size <span class="sc">|</span> qc_detected <span class="sc">|</span> qc_mito <span class="sc">|</span> qc_cell_count</span>
<span id="cb101-33"><a href="clustering.html#cb101-33" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(spe)<span class="sc">$</span>discard <span class="ot">&lt;-</span> discard</span>
<span id="cb101-34"><a href="clustering.html#cb101-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-35"><a href="clustering.html#cb101-35" aria-hidden="true" tabindex="-1"></a><span class="co"># filter low-quality spots</span></span>
<span id="cb101-36"><a href="clustering.html#cb101-36" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> spe[, <span class="sc">!</span><span class="fu">colData</span>(spe)<span class="sc">$</span>discard]</span>
<span id="cb101-37"><a href="clustering.html#cb101-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-38"><a href="clustering.html#cb101-38" aria-hidden="true" tabindex="-1"></a><span class="co"># -------------</span></span>
<span id="cb101-39"><a href="clustering.html#cb101-39" aria-hidden="true" tabindex="-1"></a><span class="co"># normalization</span></span>
<span id="cb101-40"><a href="clustering.html#cb101-40" aria-hidden="true" tabindex="-1"></a><span class="co"># -------------</span></span>
<span id="cb101-41"><a href="clustering.html#cb101-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-42"><a href="clustering.html#cb101-42" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scran)</span>
<span id="cb101-43"><a href="clustering.html#cb101-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-44"><a href="clustering.html#cb101-44" aria-hidden="true" tabindex="-1"></a><span class="co"># quick clustering for pool-based size factors</span></span>
<span id="cb101-45"><a href="clustering.html#cb101-45" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb101-46"><a href="clustering.html#cb101-46" aria-hidden="true" tabindex="-1"></a>qclus <span class="ot">&lt;-</span> <span class="fu">quickCluster</span>(spe)</span>
<span id="cb101-47"><a href="clustering.html#cb101-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-48"><a href="clustering.html#cb101-48" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate size factors and store in object</span></span>
<span id="cb101-49"><a href="clustering.html#cb101-49" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> <span class="fu">computeSumFactors</span>(spe, <span class="at">cluster =</span> qclus)</span>
<span id="cb101-50"><a href="clustering.html#cb101-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-51"><a href="clustering.html#cb101-51" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate logcounts (log-transformed normalized counts) and store in object</span></span>
<span id="cb101-52"><a href="clustering.html#cb101-52" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> <span class="fu">logNormCounts</span>(spe)</span>
<span id="cb101-53"><a href="clustering.html#cb101-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-54"><a href="clustering.html#cb101-54" aria-hidden="true" tabindex="-1"></a><span class="co"># -----------------</span></span>
<span id="cb101-55"><a href="clustering.html#cb101-55" aria-hidden="true" tabindex="-1"></a><span class="co"># feature selection</span></span>
<span id="cb101-56"><a href="clustering.html#cb101-56" aria-hidden="true" tabindex="-1"></a><span class="co"># -----------------</span></span>
<span id="cb101-57"><a href="clustering.html#cb101-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-58"><a href="clustering.html#cb101-58" aria-hidden="true" tabindex="-1"></a><span class="co"># remove mitochondrial genes</span></span>
<span id="cb101-59"><a href="clustering.html#cb101-59" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> spe[<span class="sc">!</span>is_mito, ]</span>
<span id="cb101-60"><a href="clustering.html#cb101-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-61"><a href="clustering.html#cb101-61" aria-hidden="true" tabindex="-1"></a><span class="co"># fit mean-variance relationship</span></span>
<span id="cb101-62"><a href="clustering.html#cb101-62" aria-hidden="true" tabindex="-1"></a>dec <span class="ot">&lt;-</span> <span class="fu">modelGeneVar</span>(spe)</span>
<span id="cb101-63"><a href="clustering.html#cb101-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-64"><a href="clustering.html#cb101-64" aria-hidden="true" tabindex="-1"></a><span class="co"># select top HVGs</span></span>
<span id="cb101-65"><a href="clustering.html#cb101-65" aria-hidden="true" tabindex="-1"></a>top_hvgs <span class="ot">&lt;-</span> <span class="fu">getTopHVGs</span>(dec, <span class="at">prop =</span> <span class="fl">0.1</span>)</span>
<span id="cb101-66"><a href="clustering.html#cb101-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-67"><a href="clustering.html#cb101-67" aria-hidden="true" tabindex="-1"></a><span class="co"># ------------------------</span></span>
<span id="cb101-68"><a href="clustering.html#cb101-68" aria-hidden="true" tabindex="-1"></a><span class="co"># dimensionality reduction</span></span>
<span id="cb101-69"><a href="clustering.html#cb101-69" aria-hidden="true" tabindex="-1"></a><span class="co"># ------------------------</span></span>
<span id="cb101-70"><a href="clustering.html#cb101-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-71"><a href="clustering.html#cb101-71" aria-hidden="true" tabindex="-1"></a><span class="co"># compute PCA</span></span>
<span id="cb101-72"><a href="clustering.html#cb101-72" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb101-73"><a href="clustering.html#cb101-73" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(spe, <span class="at">subset_row =</span> top_hvgs)</span>
<span id="cb101-74"><a href="clustering.html#cb101-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-75"><a href="clustering.html#cb101-75" aria-hidden="true" tabindex="-1"></a><span class="co"># compute UMAP on top 50 PCs</span></span>
<span id="cb101-76"><a href="clustering.html#cb101-76" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb101-77"><a href="clustering.html#cb101-77" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(spe, <span class="at">dimred =</span> <span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb101-78"><a href="clustering.html#cb101-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-79"><a href="clustering.html#cb101-79" aria-hidden="true" tabindex="-1"></a><span class="co"># update column names for plotting functions</span></span>
<span id="cb101-80"><a href="clustering.html#cb101-80" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(<span class="fu">reducedDim</span>(spe, <span class="st">&quot;UMAP&quot;</span>)) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;UMAP&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="clustering-on-hvgs" class="section level2" number="13.3">
<h2><span class="header-section-number">13.3</span> Clustering on HVGs</h2>
<p>We can perform clustering by applying standard clustering methods developed for single-cell RNA sequencing data, using molecular features (gene expression). Here, we apply graph-based clustering using the Walktrap method implemented in <code>scran</code> <span class="citation">(<a href="references.html#ref-Lun2016" role="doc-biblioref">Lun, McCarthy, and Marioni 2016</a>)</span>, applied to the top 50 PCs calculated on the set of top HVGs.</p>
<p>In the context of spatial data, this means we assume that biologically informative spatial distribution patterns of cell types can be detected from the molecular features (gene expression). Below, we also discuss alternative methods that directly take spatial information into account during clustering.</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="clustering.html#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="co"># graph-based clustering</span></span>
<span id="cb102-2"><a href="clustering.html#cb102-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb102-3"><a href="clustering.html#cb102-3" aria-hidden="true" tabindex="-1"></a>k <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb102-4"><a href="clustering.html#cb102-4" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">buildSNNGraph</span>(spe, <span class="at">k =</span> k, <span class="at">use.dimred =</span> <span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb102-5"><a href="clustering.html#cb102-5" aria-hidden="true" tabindex="-1"></a>g_walk <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">cluster_walktrap</span>(g)</span>
<span id="cb102-6"><a href="clustering.html#cb102-6" aria-hidden="true" tabindex="-1"></a>clus <span class="ot">&lt;-</span> g_walk<span class="sc">$</span>membership</span>
<span id="cb102-7"><a href="clustering.html#cb102-7" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(clus)</span></code></pre></div>
<pre><code>## clus
##    1    2    3    4    5    6 
##  372  916  342 1083  349  520</code></pre>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="clustering.html#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co"># store cluster labels in column &#39;label&#39; in colData</span></span>
<span id="cb104-2"><a href="clustering.html#cb104-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colLabels</span>(spe) <span class="ot">&lt;-</span> <span class="fu">factor</span>(clus)</span></code></pre></div>
<p>Visualize the clusters by plotting in (i) spatial x-y coordinates on the tissue slide, and (ii) reduced dimension space (PCA or UMAP). We use plotting functions from the <a href="https://github.com/lmweber/ggpavis">ggspavis</a> package.</p>
<p>For reference, we also display the ground truth (manually annotated) labels available for this dataset (in spatial x-y coordinates).</p>
<p>From the visualizations, we can see that the clustering reproduces the known biological structure (cortical layers), although not perfectly. The clusters are also separated in UMAP space, but again not perfectly.</p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="clustering.html#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggspavis)</span>
<span id="cb105-2"><a href="clustering.html#cb105-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-3"><a href="clustering.html#cb105-3" aria-hidden="true" tabindex="-1"></a><span class="co"># plot clusters in spatial x-y coordinates</span></span>
<span id="cb105-4"><a href="clustering.html#cb105-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSpots</span>(spe, <span class="at">discrete =</span> <span class="st">&quot;label&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;libd_layer_colors&quot;</span>)</span></code></pre></div>
<p><img src="clustering_files/figure-html/clustering_plots-1.png" width="432" /></p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="clustering.html#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot ground truth labels in spatial x-y coordinates</span></span>
<span id="cb106-2"><a href="clustering.html#cb106-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSpots</span>(spe, <span class="at">discrete =</span> <span class="st">&quot;ground_truth&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;libd_layer_colors&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: Removed 14 rows containing missing values (geom_point).</code></pre>
<p><img src="clustering_files/figure-html/clustering_plots-2.png" width="432" /></p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="clustering.html#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot clusters in PCA reduced dimensions</span></span>
<span id="cb108-2"><a href="clustering.html#cb108-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDimRed</span>(spe, <span class="at">type =</span> <span class="st">&quot;PCA&quot;</span>, <span class="at">discrete =</span> <span class="st">&quot;label&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;libd_layer_colors&quot;</span>)</span></code></pre></div>
<p><img src="clustering_files/figure-html/clustering_plots-3.png" width="432" /></p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="clustering.html#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot clusters in UMAP reduced dimensions</span></span>
<span id="cb109-2"><a href="clustering.html#cb109-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDimRed</span>(spe, <span class="at">type =</span> <span class="st">&quot;UMAP&quot;</span>, <span class="at">discrete =</span> <span class="st">&quot;label&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;libd_layer_colors&quot;</span>)</span></code></pre></div>
<p><img src="clustering_files/figure-html/clustering_plots-4.png" width="432" /></p>
</div>
<div id="clustering-on-hvgs-and-spatial-dimensions" class="section level2" number="13.4">
<h2><span class="header-section-number">13.4</span> Clustering on HVGs and spatial dimensions</h2>
<p>Section on approach from our publication <span class="citation"><a href="references.html#ref-Maynard2021" role="doc-biblioref">Maynard et al.</a> (<a href="references.html#ref-Maynard2021" role="doc-biblioref">2021</a>)</span> to combine molecular and spatial features with appropriate scaling.</p>
</div>
<div id="bayesspace" class="section level2" number="13.5">
<h2><span class="header-section-number">13.5</span> BayesSpace</h2>
<p>Section on BayesSpace <span class="citation">(<a href="references.html#ref-Zhao2020" role="doc-biblioref">Zhao et al. 2020</a>)</span>; available in Bioconductor</p>
</div>
<div id="further-methods-1" class="section level2" number="13.6">
<h2><span class="header-section-number">13.6</span> Further methods</h2>
<p>Further methods</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="dimensionality-reduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="marker-genes.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OSTA-book.pdf", "OSTA-book.epub"],
"toc": {
"collapse": "section"
},
"lib_dir": "book_assets"
});
});
</script>

</body>

</html>
