<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 16 Human DLPFC workflow | Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor</title>
  <meta name="description" content="Online textbook on ‘Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 16 Human DLPFC workflow | Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Online textbook on ‘Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor’" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 16 Human DLPFC workflow | Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor" />
  
  <meta name="twitter:description" content="Online textbook on ‘Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor’" />
  



<meta name="date" content="2021-03-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="workflows.html"/>
<link rel="next" href="mouse-coronal-workflow.html"/>
<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#authors"><i class="fa fa-check"></i><b>0.1</b> Authors</a></li>
</ul></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#contents-of-this-book"><i class="fa fa-check"></i><b>1.1</b> Contents of this book</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#who-this-book-is-for"><i class="fa fa-check"></i><b>1.2</b> Who this book is for</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#bioconductor"><i class="fa fa-check"></i><b>1.3</b> Bioconductor</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="spatially-resolved-transcriptomics.html"><a href="spatially-resolved-transcriptomics.html"><i class="fa fa-check"></i><b>2</b> Spatially resolved transcriptomics</a>
<ul>
<li class="chapter" data-level="2.1" data-path="spatially-resolved-transcriptomics.html"><a href="spatially-resolved-transcriptomics.html#x-genomics-visium"><i class="fa fa-check"></i><b>2.1</b> 10x Genomics Visium</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="spatially-resolved-transcriptomics.html"><a href="spatially-resolved-transcriptomics.html#slide-seqv2"><i class="fa fa-check"></i><b>2.1.1</b> Slide-seqV2</a></li>
<li class="chapter" data-level="2.1.2" data-path="spatially-resolved-transcriptomics.html"><a href="spatially-resolved-transcriptomics.html#seqfish"><i class="fa fa-check"></i><b>2.1.2</b> seqFISH</a></li>
<li class="chapter" data-level="2.1.3" data-path="spatially-resolved-transcriptomics.html"><a href="spatially-resolved-transcriptomics.html#merfish"><i class="fa fa-check"></i><b>2.1.3</b> MERFISH</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spatialexperiment.html"><a href="spatialexperiment.html"><i class="fa fa-check"></i><b>3</b> SpatialExperiment</a>
<ul>
<li class="chapter" data-level="3.1" data-path="spatialexperiment.html"><a href="spatialexperiment.html#spatialexperiment-class"><i class="fa fa-check"></i><b>3.1</b> SpatialExperiment class</a></li>
</ul></li>
<li class="part"><span><b>II Preprocessing steps</b></span></li>
<li class="chapter" data-level="4" data-path="preprocessing-steps.html"><a href="preprocessing-steps.html"><i class="fa fa-check"></i><b>4</b> Preprocessing steps</a></li>
<li class="chapter" data-level="5" data-path="space-ranger.html"><a href="space-ranger.html"><i class="fa fa-check"></i><b>5</b> Space Ranger</a>
<ul>
<li class="chapter" data-level="5.1" data-path="space-ranger.html"><a href="space-ranger.html#overview"><i class="fa fa-check"></i><b>5.1</b> Overview</a></li>
<li class="chapter" data-level="5.2" data-path="space-ranger.html"><a href="space-ranger.html#installation"><i class="fa fa-check"></i><b>5.2</b> Installation</a></li>
<li class="chapter" data-level="5.3" data-path="space-ranger.html"><a href="space-ranger.html#memory-issues"><i class="fa fa-check"></i><b>5.3</b> Memory issues</a></li>
<li class="chapter" data-level="5.4" data-path="space-ranger.html"><a href="space-ranger.html#how-to-run-spaceranger-count"><i class="fa fa-check"></i><b>5.4</b> How to run spaceranger count</a></li>
<li class="chapter" data-level="5.5" data-path="space-ranger.html"><a href="space-ranger.html#image-alignment-.json-file"><i class="fa fa-check"></i><b>5.5</b> Image alignment .json file</a></li>
<li class="chapter" data-level="5.6" data-path="space-ranger.html"><a href="space-ranger.html#output-files-and-importing-into-r"><i class="fa fa-check"></i><b>5.6</b> Output files and importing into R</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="loupe-browser.html"><a href="loupe-browser.html"><i class="fa fa-check"></i><b>6</b> Loupe Browser</a>
<ul>
<li class="chapter" data-level="6.1" data-path="loupe-browser.html"><a href="loupe-browser.html#overview-1"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="loupe-browser.html"><a href="loupe-browser.html#automatic-alignment-of-images"><i class="fa fa-check"></i><b>6.2</b> Automatic alignment of images</a></li>
<li class="chapter" data-level="6.3" data-path="loupe-browser.html"><a href="loupe-browser.html#manual-alignment-of-images"><i class="fa fa-check"></i><b>6.3</b> Manual alignment of images</a></li>
<li class="chapter" data-level="6.4" data-path="loupe-browser.html"><a href="loupe-browser.html#web-summary-.html-file"><i class="fa fa-check"></i><b>6.4</b> Web summary .html file</a></li>
<li class="chapter" data-level="6.5" data-path="loupe-browser.html"><a href="loupe-browser.html#downstream-analyses-in-loupe-browser"><i class="fa fa-check"></i><b>6.5</b> Downstream analyses in Loupe Browser</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="image-segmentation.html"><a href="image-segmentation.html"><i class="fa fa-check"></i><b>7</b> Image segmentation</a>
<ul>
<li class="chapter" data-level="7.1" data-path="image-segmentation.html"><a href="image-segmentation.html#overview-2"><i class="fa fa-check"></i><b>7.1</b> Overview</a></li>
<li class="chapter" data-level="7.2" data-path="image-segmentation.html"><a href="image-segmentation.html#processing-image-files"><i class="fa fa-check"></i><b>7.2</b> Processing image files</a></li>
<li class="chapter" data-level="7.3" data-path="image-segmentation.html"><a href="image-segmentation.html#identify-number-of-cells-per-spot"><i class="fa fa-check"></i><b>7.3</b> Identify number of cells per spot</a></li>
</ul></li>
<li class="part"><span><b>III Analysis steps</b></span></li>
<li class="chapter" data-level="8" data-path="analysis-steps.html"><a href="analysis-steps.html"><i class="fa fa-check"></i><b>8</b> Analysis steps</a>
<ul>
<li class="chapter" data-level="8.1" data-path="analysis-steps.html"><a href="analysis-steps.html#load-data"><i class="fa fa-check"></i><b>8.1</b> Load data</a></li>
<li class="chapter" data-level="8.2" data-path="analysis-steps.html"><a href="analysis-steps.html#spatialexperiment-object-structure"><i class="fa fa-check"></i><b>8.2</b> SpatialExperiment object structure</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>9</b> Quality control</a>
<ul>
<li class="chapter" data-level="9.1" data-path="quality-control.html"><a href="quality-control.html#background"><i class="fa fa-check"></i><b>9.1</b> Background</a></li>
<li class="chapter" data-level="9.2" data-path="quality-control.html"><a href="quality-control.html#load-data-1"><i class="fa fa-check"></i><b>9.2</b> Load data</a></li>
<li class="chapter" data-level="9.3" data-path="quality-control.html"><a href="quality-control.html#plot-data"><i class="fa fa-check"></i><b>9.3</b> Plot data</a></li>
<li class="chapter" data-level="9.4" data-path="quality-control.html"><a href="quality-control.html#calculate-qc-metrics"><i class="fa fa-check"></i><b>9.4</b> Calculate QC metrics</a></li>
<li class="chapter" data-level="9.5" data-path="quality-control.html"><a href="quality-control.html#selecting-thresholds"><i class="fa fa-check"></i><b>9.5</b> Selecting thresholds</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="quality-control.html"><a href="quality-control.html#library-size"><i class="fa fa-check"></i><b>9.5.1</b> Library size</a></li>
<li class="chapter" data-level="9.5.2" data-path="quality-control.html"><a href="quality-control.html#number-of-expressed-features"><i class="fa fa-check"></i><b>9.5.2</b> Number of expressed features</a></li>
<li class="chapter" data-level="9.5.3" data-path="quality-control.html"><a href="quality-control.html#proportion-of-mitochondrial-reads"><i class="fa fa-check"></i><b>9.5.3</b> Proportion of mitochondrial reads</a></li>
<li class="chapter" data-level="9.5.4" data-path="quality-control.html"><a href="quality-control.html#number-of-cells-per-spot"><i class="fa fa-check"></i><b>9.5.4</b> Number of cells per spot</a></li>
<li class="chapter" data-level="9.5.5" data-path="quality-control.html"><a href="quality-control.html#remove-low-quality-spots"><i class="fa fa-check"></i><b>9.5.5</b> Remove low-quality spots</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="quality-control.html"><a href="quality-control.html#zero-cell-and-single-cell-spots"><i class="fa fa-check"></i><b>9.6</b> Zero-cell and single-cell spots</a></li>
<li class="chapter" data-level="9.7" data-path="quality-control.html"><a href="quality-control.html#quality-control-at-gene-level"><i class="fa fa-check"></i><b>9.7</b> Quality control at gene level</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="normalization.html"><a href="normalization.html"><i class="fa fa-check"></i><b>10</b> Normalization</a>
<ul>
<li class="chapter" data-level="10.1" data-path="normalization.html"><a href="normalization.html#background-1"><i class="fa fa-check"></i><b>10.1</b> Background</a></li>
<li class="chapter" data-level="10.2" data-path="normalization.html"><a href="normalization.html#previous-steps"><i class="fa fa-check"></i><b>10.2</b> Previous steps</a></li>
<li class="chapter" data-level="10.3" data-path="normalization.html"><a href="normalization.html#log-transformed-normalized-counts"><i class="fa fa-check"></i><b>10.3</b> Log-transformed normalized counts</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="feature-selection.html"><a href="feature-selection.html"><i class="fa fa-check"></i><b>11</b> Feature selection</a>
<ul>
<li class="chapter" data-level="11.1" data-path="feature-selection.html"><a href="feature-selection.html#background-2"><i class="fa fa-check"></i><b>11.1</b> Background</a></li>
<li class="chapter" data-level="11.2" data-path="feature-selection.html"><a href="feature-selection.html#previous-steps-1"><i class="fa fa-check"></i><b>11.2</b> Previous steps</a></li>
<li class="chapter" data-level="11.3" data-path="feature-selection.html"><a href="feature-selection.html#highly-variable-genes-hvgs"><i class="fa fa-check"></i><b>11.3</b> Highly variable genes (HVGs)</a></li>
<li class="chapter" data-level="11.4" data-path="feature-selection.html"><a href="feature-selection.html#spatially-variable-genes-svgs"><i class="fa fa-check"></i><b>11.4</b> Spatially variable genes (SVGs)</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="feature-selection.html"><a href="feature-selection.html#spatialde"><i class="fa fa-check"></i><b>11.4.1</b> SpatialDE</a></li>
<li class="chapter" data-level="11.4.2" data-path="feature-selection.html"><a href="feature-selection.html#spark"><i class="fa fa-check"></i><b>11.4.2</b> SPARK</a></li>
<li class="chapter" data-level="11.4.3" data-path="feature-selection.html"><a href="feature-selection.html#morans-i-statistic"><i class="fa fa-check"></i><b>11.4.3</b> Moran’s I statistic</a></li>
<li class="chapter" data-level="11.4.4" data-path="feature-selection.html"><a href="feature-selection.html#further-methods"><i class="fa fa-check"></i><b>11.4.4</b> Further methods</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="dimensionality-reduction.html"><a href="dimensionality-reduction.html"><i class="fa fa-check"></i><b>12</b> Dimensionality reduction</a>
<ul>
<li class="chapter" data-level="12.1" data-path="dimensionality-reduction.html"><a href="dimensionality-reduction.html#background-3"><i class="fa fa-check"></i><b>12.1</b> Background</a></li>
<li class="chapter" data-level="12.2" data-path="dimensionality-reduction.html"><a href="dimensionality-reduction.html#previous-steps-2"><i class="fa fa-check"></i><b>12.2</b> Previous steps</a></li>
<li class="chapter" data-level="12.3" data-path="dimensionality-reduction.html"><a href="dimensionality-reduction.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>12.3</b> Principal component analysis (PCA)</a></li>
<li class="chapter" data-level="12.4" data-path="dimensionality-reduction.html"><a href="dimensionality-reduction.html#uniform-manifold-approximation-and-projection-umap"><i class="fa fa-check"></i><b>12.4</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="12.5" data-path="dimensionality-reduction.html"><a href="dimensionality-reduction.html#visualizations"><i class="fa fa-check"></i><b>12.5</b> Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="clustering.html"><a href="clustering.html"><i class="fa fa-check"></i><b>13</b> Clustering</a>
<ul>
<li class="chapter" data-level="13.1" data-path="clustering.html"><a href="clustering.html#overview-3"><i class="fa fa-check"></i><b>13.1</b> Overview</a></li>
<li class="chapter" data-level="13.2" data-path="clustering.html"><a href="clustering.html#previous-steps-3"><i class="fa fa-check"></i><b>13.2</b> Previous steps</a></li>
<li class="chapter" data-level="13.3" data-path="clustering.html"><a href="clustering.html#clustering-on-hvgs"><i class="fa fa-check"></i><b>13.3</b> Clustering on HVGs</a></li>
<li class="chapter" data-level="13.4" data-path="clustering.html"><a href="clustering.html#clustering-on-hvgs-and-spatial-dimensions"><i class="fa fa-check"></i><b>13.4</b> Clustering on HVGs and spatial dimensions</a></li>
<li class="chapter" data-level="13.5" data-path="clustering.html"><a href="clustering.html#bayesspace"><i class="fa fa-check"></i><b>13.5</b> BayesSpace</a></li>
<li class="chapter" data-level="13.6" data-path="clustering.html"><a href="clustering.html#further-methods-1"><i class="fa fa-check"></i><b>13.6</b> Further methods</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="marker-genes.html"><a href="marker-genes.html"><i class="fa fa-check"></i><b>14</b> Marker genes</a>
<ul>
<li class="chapter" data-level="14.1" data-path="marker-genes.html"><a href="marker-genes.html#background-4"><i class="fa fa-check"></i><b>14.1</b> Background</a></li>
<li class="chapter" data-level="14.2" data-path="marker-genes.html"><a href="marker-genes.html#previous-steps-4"><i class="fa fa-check"></i><b>14.2</b> Previous steps</a></li>
<li class="chapter" data-level="14.3" data-path="marker-genes.html"><a href="marker-genes.html#marker-genes"><i class="fa fa-check"></i><b>14.3</b> Marker genes</a></li>
</ul></li>
<li class="part"><span><b>IV Workflows</b></span></li>
<li class="chapter" data-level="15" data-path="workflows.html"><a href="workflows.html"><i class="fa fa-check"></i><b>15</b> Workflows</a></li>
<li class="chapter" data-level="16" data-path="human-DLPFC-workflow.html"><a href="human-DLPFC-workflow.html"><i class="fa fa-check"></i><b>16</b> Human DLPFC workflow</a>
<ul>
<li class="chapter" data-level="16.1" data-path="human-DLPFC-workflow.html"><a href="human-DLPFC-workflow.html#description-of-dataset"><i class="fa fa-check"></i><b>16.1</b> Description of dataset</a></li>
<li class="chapter" data-level="16.2" data-path="human-DLPFC-workflow.html"><a href="human-DLPFC-workflow.html#load-data-2"><i class="fa fa-check"></i><b>16.2</b> Load data</a></li>
<li class="chapter" data-level="16.3" data-path="human-DLPFC-workflow.html"><a href="human-DLPFC-workflow.html#plot-data-1"><i class="fa fa-check"></i><b>16.3</b> Plot data</a></li>
<li class="chapter" data-level="16.4" data-path="human-DLPFC-workflow.html"><a href="human-DLPFC-workflow.html#quality-control-qc"><i class="fa fa-check"></i><b>16.4</b> Quality control (QC)</a></li>
<li class="chapter" data-level="16.5" data-path="human-DLPFC-workflow.html"><a href="human-DLPFC-workflow.html#normalization-1"><i class="fa fa-check"></i><b>16.5</b> Normalization</a></li>
<li class="chapter" data-level="16.6" data-path="human-DLPFC-workflow.html"><a href="human-DLPFC-workflow.html#feature-selection"><i class="fa fa-check"></i><b>16.6</b> Feature selection</a></li>
<li class="chapter" data-level="16.7" data-path="human-DLPFC-workflow.html"><a href="human-DLPFC-workflow.html#dimensionality-reduction"><i class="fa fa-check"></i><b>16.7</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="16.8" data-path="human-DLPFC-workflow.html"><a href="human-DLPFC-workflow.html#clustering-1"><i class="fa fa-check"></i><b>16.8</b> Clustering</a></li>
<li class="chapter" data-level="16.9" data-path="human-DLPFC-workflow.html"><a href="human-DLPFC-workflow.html#marker-genes-1"><i class="fa fa-check"></i><b>16.9</b> Marker genes</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html"><i class="fa fa-check"></i><b>17</b> Mouse coronal workflow</a>
<ul>
<li class="chapter" data-level="17.1" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html#description-of-dataset-1"><i class="fa fa-check"></i><b>17.1</b> Description of dataset</a></li>
<li class="chapter" data-level="17.2" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html#load-data-3"><i class="fa fa-check"></i><b>17.2</b> Load data</a></li>
<li class="chapter" data-level="17.3" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html#plot-data-2"><i class="fa fa-check"></i><b>17.3</b> Plot data</a></li>
<li class="chapter" data-level="17.4" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html#quality-control-qc-1"><i class="fa fa-check"></i><b>17.4</b> Quality control (QC)</a></li>
<li class="chapter" data-level="17.5" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html#normalization-2"><i class="fa fa-check"></i><b>17.5</b> Normalization</a></li>
<li class="chapter" data-level="17.6" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html#feature-selection-1"><i class="fa fa-check"></i><b>17.6</b> Feature selection</a></li>
<li class="chapter" data-level="17.7" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html#dimensionality-reduction-1"><i class="fa fa-check"></i><b>17.7</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="17.8" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html#clustering-2"><i class="fa fa-check"></i><b>17.8</b> Clustering</a></li>
<li class="chapter" data-level="17.9" data-path="mouse-coronal-workflow.html"><a href="mouse-coronal-workflow.html#marker-genes-2"><i class="fa fa-check"></i><b>17.9</b> Marker genes</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="human_DLPFC_workflow" class="section level1" number="16">
<h1><span class="header-section-number">Chapter 16</span> Human DLPFC workflow</h1>
<p>This workflow analyzes one sample of human brain from the dorsolateral prefrontal cortex (DLPFC) region, measured using the 10x Genomics Visium platform. This is a condensed version of the analyses shown in the individual analysis chapters in the previous part. For more details on any of the individual steps, see the previous chapters.</p>
<div id="description-of-dataset" class="section level2" number="16.1">
<h2><span class="header-section-number">16.1</span> Description of dataset</h2>
<p>This is a 10x Genomics Visium dataset generated from healthy human brain samples from the dorsolateral prefrontal cortex (DLPFC) region.</p>
<p>In the full dataset, there are 12 samples in total, from 3 individuals, with 2 pairs of “spatially adjacent” replicates (serial sections) per individual (4 samples per individual). The individuals and spatially adjacent replicates can be considered as blocking factors. Each sample spans the six layers of the cortex plus white matter in a perpendicular tissue section.</p>
<p>For the examples in this workflow and the analysis chapters, we use a single sample from this dataset (sample 151673), mainly to reduce computational requirements to compile the book.</p>
<p>This dataset is described in our publication <span class="citation"><a href="mouse-coronal-workflow.html#ref-Maynard2020-ke" role="doc-biblioref">Maynard et al.</a> (<a href="mouse-coronal-workflow.html#ref-Maynard2020-ke" role="doc-biblioref">2021</a>)</span>. The full dataset is publicly available through the <a href="http://bioconductor.org/packages/spatialLIBD">spatialLIBD</a> Bioconductor package, and the analysis code from our paper is provided in the <a href="https://github.com/LieberInstitute/HumanPilot">HumanPilot</a> GitHub repository. The dataset can also be explored interactively through the <a href="http://spatial.libd.org/spatialLIBD/">spatialLIBD Shiny web app</a>.</p>
</div>
<div id="load-data-2" class="section level2" number="16.2">
<h2><span class="header-section-number">16.2</span> Load data</h2>
<p>Here, we load a single sample from this dataset (sample 151673), which is available as a <code>SpatialExperiment</code> object from the <a href="https://github.com/lmweber/STexampleData">STexampleData</a> package.</p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="human-DLPFC-workflow.html#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SpatialExperiment)</span>
<span id="cb105-2"><a href="human-DLPFC-workflow.html#cb105-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(STexampleData)</span>
<span id="cb105-3"><a href="human-DLPFC-workflow.html#cb105-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-4"><a href="human-DLPFC-workflow.html#cb105-4" aria-hidden="true" tabindex="-1"></a><span class="co"># load object</span></span>
<span id="cb105-5"><a href="human-DLPFC-workflow.html#cb105-5" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> <span class="fu">load_data</span>(<span class="st">&quot;Visium_humanDLPFC&quot;</span>)</span>
<span id="cb105-6"><a href="human-DLPFC-workflow.html#cb105-6" aria-hidden="true" tabindex="-1"></a>spe</span></code></pre></div>
<pre><code>## class: SpatialExperiment 
## dim: 33538 4992 
## metadata(0):
## assays(1): counts
## rownames(33538): ENSG00000243485 ENSG00000237613 ... ENSG00000277475
##   ENSG00000268674
## rowData names(3): gene_id gene_name feature_type
## colnames(4992): AAACAACGAATAGTTC-1 AAACAAGTATCTCCCA-1 ...
##   TTGTTTGTATTACACG-1 TTGTTTGTGTAAATTC-1
## colData names(10): barcode_id imagerow ... pxl_col_in_fullres
##   pxl_row_in_fullres
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## spatialData names(4) : barcode_id in_tissue x y
## imgData names(4): sample_id image_id data scaleFactor</code></pre>
</div>
<div id="plot-data-1" class="section level2" number="16.3">
<h2><span class="header-section-number">16.3</span> Plot data</h2>
<p>As an initial check, plot the spatial coordinates (spots) in x-y dimensions on the tissue slide, to check that the object has loaded correctly and that the orientation is as expected.</p>
<p>We use visualization functions from the <a href="https://github.com/lmweber/ggpavis">ggspavis</a> package to generate plots.</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="human-DLPFC-workflow.html#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggspavis)</span>
<span id="cb107-2"><a href="human-DLPFC-workflow.html#cb107-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-3"><a href="human-DLPFC-workflow.html#cb107-3" aria-hidden="true" tabindex="-1"></a><span class="co"># plot spatial coordinates (spots)</span></span>
<span id="cb107-4"><a href="human-DLPFC-workflow.html#cb107-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSpots</span>(spe)</span></code></pre></div>
<p><img src="workflow_human_DLPFC_files/figure-html/plot_data-1.png" width="360" /></p>
</div>
<div id="quality-control-qc" class="section level2" number="16.4">
<h2><span class="header-section-number">16.4</span> Quality control (QC)</h2>
<p>First, we subset the object to keep only spots over tissue. The remaining spots are background spots, which we are not interested in.</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="human-DLPFC-workflow.html#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="co"># subset to keep only spots over tissue</span></span>
<span id="cb108-2"><a href="human-DLPFC-workflow.html#cb108-2" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> spe[, <span class="fu">spatialData</span>(spe)<span class="sc">$</span>in_tissue <span class="sc">==</span> <span class="dv">1</span>]</span>
<span id="cb108-3"><a href="human-DLPFC-workflow.html#cb108-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(spe)</span></code></pre></div>
<pre><code>## [1] 33538  3639</code></pre>
<p>Next, calculate spot-level QC metrics using the <code>scater</code> package <span class="citation">(<a href="mouse-coronal-workflow.html#ref-McCarthy2017-zd" role="doc-biblioref">McCarthy et al. 2017</a>)</span>, and store the QC metrics in <code>colData</code>. See <a href="quality-control.html#quality_control">Quality control</a> for more details, including explanations of the QC metrics.</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="human-DLPFC-workflow.html#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb110-2"><a href="human-DLPFC-workflow.html#cb110-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-3"><a href="human-DLPFC-workflow.html#cb110-3" aria-hidden="true" tabindex="-1"></a><span class="co"># identify mitochondrial genes</span></span>
<span id="cb110-4"><a href="human-DLPFC-workflow.html#cb110-4" aria-hidden="true" tabindex="-1"></a>is_mito <span class="ot">&lt;-</span> <span class="fu">grepl</span>(<span class="st">&quot;(^MT-)|(^mt-)&quot;</span>, <span class="fu">rowData</span>(spe)<span class="sc">$</span>gene_name)</span>
<span id="cb110-5"><a href="human-DLPFC-workflow.html#cb110-5" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(is_mito)</span></code></pre></div>
<pre><code>## is_mito
## FALSE  TRUE 
## 33525    13</code></pre>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="human-DLPFC-workflow.html#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(spe)<span class="sc">$</span>gene_name[is_mito]</span></code></pre></div>
<pre><code>##  [1] &quot;MT-ND1&quot;  &quot;MT-ND2&quot;  &quot;MT-CO1&quot;  &quot;MT-CO2&quot;  &quot;MT-ATP8&quot; &quot;MT-ATP6&quot; &quot;MT-CO3&quot; 
##  [8] &quot;MT-ND3&quot;  &quot;MT-ND4L&quot; &quot;MT-ND4&quot;  &quot;MT-ND5&quot;  &quot;MT-ND6&quot;  &quot;MT-CYB&quot;</code></pre>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="human-DLPFC-workflow.html#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate per-spot QC metrics and store in colData</span></span>
<span id="cb114-2"><a href="human-DLPFC-workflow.html#cb114-2" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> <span class="fu">addPerCellQC</span>(spe, <span class="at">subsets =</span> <span class="fu">list</span>(<span class="at">mito =</span> is_mito))</span>
<span id="cb114-3"><a href="human-DLPFC-workflow.html#cb114-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">colData</span>(spe), <span class="dv">3</span>)</span></code></pre></div>
<pre><code>## DataFrame with 3 rows and 16 columns
##                            barcode_id  imagerow  imagecol cell_count
##                           &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt;  &lt;integer&gt;
## AAACAAGTATCTCCCA-1 AAACAAGTATCTCCCA-1   381.098   440.639          6
## AAACAATCTACTAGCA-1 AAACAATCTACTAGCA-1   126.328   259.631         16
## AAACACCAATAACTGC-1 AAACACCAATAACTGC-1   427.768   183.078          5
##                    ground_truth     sample_id array_row array_col
##                        &lt;factor&gt;   &lt;character&gt; &lt;integer&gt; &lt;integer&gt;
## AAACAAGTATCTCCCA-1       Layer3 sample_151673        50       102
## AAACAATCTACTAGCA-1       Layer1 sample_151673         3        43
## AAACACCAATAACTGC-1       WM     sample_151673        59        19
##                    pxl_col_in_fullres pxl_row_in_fullres       sum  detected
##                             &lt;integer&gt;          &lt;integer&gt; &lt;numeric&gt; &lt;numeric&gt;
## AAACAAGTATCTCCCA-1               8468               9791      8458      3586
## AAACAATCTACTAGCA-1               2807               5769      1667      1150
## AAACACCAATAACTGC-1               9505               4068      3769      1960
##                    subsets_mito_sum subsets_mito_detected subsets_mito_percent
##                           &lt;numeric&gt;             &lt;numeric&gt;            &lt;numeric&gt;
## AAACAAGTATCTCCCA-1             1407                    13              16.6351
## AAACAATCTACTAGCA-1              204                    11              12.2376
## AAACACCAATAACTGC-1              430                    13              11.4089
##                        total
##                    &lt;numeric&gt;
## AAACAAGTATCTCCCA-1      8458
## AAACAATCTACTAGCA-1      1667
## AAACACCAATAACTGC-1      3769</code></pre>
<p>Select filtering thresholds for the QC metrics by examining distributions using histograms. For additional details, including further exploratory visualizations to select the thresholds, see <a href="quality-control.html#quality_control">Quality control</a>. Here, we use relatively relaxed thresholds, since the additional exploratory visualizations showed that more stringent thresholds tended to remove groups of spots corresponding to biologically meaningful regions.</p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="human-DLPFC-workflow.html#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="co"># histograms of QC metrics</span></span>
<span id="cb116-2"><a href="human-DLPFC-workflow.html#cb116-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">4</span>))</span>
<span id="cb116-3"><a href="human-DLPFC-workflow.html#cb116-3" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">colData</span>(spe)<span class="sc">$</span>sum, <span class="at">xlab =</span> <span class="st">&quot;sum&quot;</span>, <span class="at">main =</span> <span class="st">&quot;UMIs per spot&quot;</span>)</span>
<span id="cb116-4"><a href="human-DLPFC-workflow.html#cb116-4" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">colData</span>(spe)<span class="sc">$</span>detected, <span class="at">xlab =</span> <span class="st">&quot;detected&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Genes per spot&quot;</span>)</span>
<span id="cb116-5"><a href="human-DLPFC-workflow.html#cb116-5" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">colData</span>(spe)<span class="sc">$</span>subsets_mito_percent, <span class="at">xlab =</span> <span class="st">&quot;percent mitochondrial&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Percent mito UMIs&quot;</span>)</span>
<span id="cb116-6"><a href="human-DLPFC-workflow.html#cb116-6" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">colData</span>(spe)<span class="sc">$</span>cell_count, <span class="at">xlab =</span> <span class="st">&quot;number of cells&quot;</span>, <span class="at">main =</span> <span class="st">&quot;No. cells per spot&quot;</span>)</span></code></pre></div>
<p><img src="workflow_human_DLPFC_files/figure-html/QC_thresholds-1.png" width="720" /></p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="human-DLPFC-workflow.html#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb117-2"><a href="human-DLPFC-workflow.html#cb117-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-3"><a href="human-DLPFC-workflow.html#cb117-3" aria-hidden="true" tabindex="-1"></a><span class="co"># select QC thresholds</span></span>
<span id="cb117-4"><a href="human-DLPFC-workflow.html#cb117-4" aria-hidden="true" tabindex="-1"></a>qc_lib_size <span class="ot">&lt;-</span> <span class="fu">colData</span>(spe)<span class="sc">$</span>sum <span class="sc">&lt;</span> <span class="dv">500</span></span>
<span id="cb117-5"><a href="human-DLPFC-workflow.html#cb117-5" aria-hidden="true" tabindex="-1"></a>qc_detected <span class="ot">&lt;-</span> <span class="fu">colData</span>(spe)<span class="sc">$</span>detected <span class="sc">&lt;</span> <span class="dv">250</span></span>
<span id="cb117-6"><a href="human-DLPFC-workflow.html#cb117-6" aria-hidden="true" tabindex="-1"></a>qc_mito <span class="ot">&lt;-</span> <span class="fu">colData</span>(spe)<span class="sc">$</span>subsets_mito_percent <span class="sc">&gt;</span> <span class="dv">30</span></span>
<span id="cb117-7"><a href="human-DLPFC-workflow.html#cb117-7" aria-hidden="true" tabindex="-1"></a>qc_cell_count <span class="ot">&lt;-</span> <span class="fu">colData</span>(spe)<span class="sc">$</span>cell_count <span class="sc">&gt;</span> <span class="dv">12</span></span>
<span id="cb117-8"><a href="human-DLPFC-workflow.html#cb117-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-9"><a href="human-DLPFC-workflow.html#cb117-9" aria-hidden="true" tabindex="-1"></a><span class="co"># number of discarded spots for each metric</span></span>
<span id="cb117-10"><a href="human-DLPFC-workflow.html#cb117-10" aria-hidden="true" tabindex="-1"></a><span class="fu">apply</span>(<span class="fu">cbind</span>(qc_lib_size, qc_detected, qc_mito, qc_cell_count), <span class="dv">2</span>, sum)</span></code></pre></div>
<pre><code>##   qc_lib_size   qc_detected       qc_mito qc_cell_count 
##             7             5             3            47</code></pre>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="human-DLPFC-workflow.html#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="co"># combined set of discarded spots</span></span>
<span id="cb119-2"><a href="human-DLPFC-workflow.html#cb119-2" aria-hidden="true" tabindex="-1"></a>discard <span class="ot">&lt;-</span> qc_lib_size <span class="sc">|</span> qc_detected <span class="sc">|</span> qc_mito <span class="sc">|</span> qc_cell_count</span>
<span id="cb119-3"><a href="human-DLPFC-workflow.html#cb119-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(discard)</span></code></pre></div>
<pre><code>## discard
## FALSE  TRUE 
##  3582    57</code></pre>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="human-DLPFC-workflow.html#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="co"># store in object</span></span>
<span id="cb121-2"><a href="human-DLPFC-workflow.html#cb121-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(spe)<span class="sc">$</span>discard <span class="ot">&lt;-</span> discard</span></code></pre></div>
<p>Plot the set of discarded spots in the spatial x-y coordinates, to confirm that the spatial distribution of the discarded spots does not correspond to any biologically meaningful regions, which would indicate that we are removing biologically informative spots.</p>
<p>We use QC visualization functions from the <a href="https://github.com/lmweber/spatzli">spatzli</a> package to generate plots.</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="human-DLPFC-workflow.html#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(spatzli)</span>
<span id="cb122-2"><a href="human-DLPFC-workflow.html#cb122-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-3"><a href="human-DLPFC-workflow.html#cb122-3" aria-hidden="true" tabindex="-1"></a><span class="co"># check spatial pattern of discarded spots</span></span>
<span id="cb122-4"><a href="human-DLPFC-workflow.html#cb122-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQCspots</span>(spe, <span class="at">discard =</span> <span class="st">&quot;discard&quot;</span>)</span></code></pre></div>
<p><img src="workflow_human_DLPFC_files/figure-html/QC_check-1.png" width="672" /></p>
<p>There is some concentration of discarded spots at the edge of the tissue region, which may be due to tissue damage. Importantly, the discarded spots do not correspond to any of the cortical layers of interest.</p>
<p>We filter out the low-quality spots from the object.</p>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="human-DLPFC-workflow.html#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="co"># filter low-quality spots</span></span>
<span id="cb123-2"><a href="human-DLPFC-workflow.html#cb123-2" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> spe[, <span class="sc">!</span><span class="fu">colData</span>(spe)<span class="sc">$</span>discard]</span>
<span id="cb123-3"><a href="human-DLPFC-workflow.html#cb123-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(spe)</span></code></pre></div>
<pre><code>## [1] 33538  3582</code></pre>
</div>
<div id="normalization-1" class="section level2" number="16.5">
<h2><span class="header-section-number">16.5</span> Normalization</h2>
<p>Calculate log-transformed normalized counts, using pool-based size factors and deconvolution to the spot level. We use normalization methods from <code>scater</code> <span class="citation">(<a href="mouse-coronal-workflow.html#ref-McCarthy2017-zd" role="doc-biblioref">McCarthy et al. 2017</a>)</span> and <code>scran</code> <span class="citation">(<a href="mouse-coronal-workflow.html#ref-Lun2016-dn" role="doc-biblioref">Lun, McCarthy, and Marioni 2016</a>)</span>, by assuming that these methods can be applied by treating spots as equivalent to cells. Since we have a single sample, there are no blocking factors. For more details, see <a href="normalization.html#normalization">Normalization</a>.</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="human-DLPFC-workflow.html#cb125-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scran)</span>
<span id="cb125-2"><a href="human-DLPFC-workflow.html#cb125-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-3"><a href="human-DLPFC-workflow.html#cb125-3" aria-hidden="true" tabindex="-1"></a><span class="co"># quick clustering for pool-based size factors</span></span>
<span id="cb125-4"><a href="human-DLPFC-workflow.html#cb125-4" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb125-5"><a href="human-DLPFC-workflow.html#cb125-5" aria-hidden="true" tabindex="-1"></a>qclus <span class="ot">&lt;-</span> <span class="fu">quickCluster</span>(spe)</span>
<span id="cb125-6"><a href="human-DLPFC-workflow.html#cb125-6" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(qclus)</span></code></pre></div>
<pre><code>## qclus
##   1   2   3   4   5   6   7   8   9  10 
## 372 245 254 744 415 230 394 299 492 137</code></pre>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="human-DLPFC-workflow.html#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate size factors and store in object</span></span>
<span id="cb127-2"><a href="human-DLPFC-workflow.html#cb127-2" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> <span class="fu">computeSumFactors</span>(spe, <span class="at">cluster =</span> qclus)</span>
<span id="cb127-3"><a href="human-DLPFC-workflow.html#cb127-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-4"><a href="human-DLPFC-workflow.html#cb127-4" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">sizeFactors</span>(spe))</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.1334  0.6093  0.8844  1.0000  1.2852  4.2475</code></pre>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="human-DLPFC-workflow.html#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">sizeFactors</span>(spe), <span class="at">breaks =</span> <span class="dv">20</span>)</span></code></pre></div>
<p><img src="workflow_human_DLPFC_files/figure-html/normalization-1.png" width="672" /></p>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="human-DLPFC-workflow.html#cb130-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate logcounts (log-transformed normalized counts) and store in object</span></span>
<span id="cb130-2"><a href="human-DLPFC-workflow.html#cb130-2" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> <span class="fu">logNormCounts</span>(spe)</span>
<span id="cb130-3"><a href="human-DLPFC-workflow.html#cb130-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-4"><a href="human-DLPFC-workflow.html#cb130-4" aria-hidden="true" tabindex="-1"></a><span class="fu">assayNames</span>(spe)</span></code></pre></div>
<pre><code>## [1] &quot;counts&quot;    &quot;logcounts&quot;</code></pre>
</div>
<div id="feature-selection" class="section level2" number="16.6">
<h2><span class="header-section-number">16.6</span> Feature selection</h2>
<p>Identify a set of top highly variable genes (HVGs), which will be used to define cell types. We use methods from <code>scran</code> <span class="citation">(<a href="mouse-coronal-workflow.html#ref-Lun2016-dn" role="doc-biblioref">Lun, McCarthy, and Marioni 2016</a>)</span>, treating spots as equivalent to cells, and considering only molecular features (gene expression). Alternative methods adapted for spatial data will be described in <a href="feature-selection.html#feature_selection">Feature selection</a>. We also first filter out mitochondrial genes, since these are very highly expressed and not of biological interest here.</p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="human-DLPFC-workflow.html#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="co"># remove mitochondrial genes</span></span>
<span id="cb132-2"><a href="human-DLPFC-workflow.html#cb132-2" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> spe[<span class="sc">!</span>is_mito, ]</span>
<span id="cb132-3"><a href="human-DLPFC-workflow.html#cb132-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(spe)</span></code></pre></div>
<pre><code>## [1] 33525  3582</code></pre>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="human-DLPFC-workflow.html#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="co"># fit mean-variance relationship</span></span>
<span id="cb134-2"><a href="human-DLPFC-workflow.html#cb134-2" aria-hidden="true" tabindex="-1"></a>dec <span class="ot">&lt;-</span> <span class="fu">modelGeneVar</span>(spe)</span>
<span id="cb134-3"><a href="human-DLPFC-workflow.html#cb134-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-4"><a href="human-DLPFC-workflow.html#cb134-4" aria-hidden="true" tabindex="-1"></a><span class="co"># visualize mean-variance relationship</span></span>
<span id="cb134-5"><a href="human-DLPFC-workflow.html#cb134-5" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">metadata</span>(dec)</span>
<span id="cb134-6"><a href="human-DLPFC-workflow.html#cb134-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(fit<span class="sc">$</span>mean, fit<span class="sc">$</span>var, </span>
<span id="cb134-7"><a href="human-DLPFC-workflow.html#cb134-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;mean of log-expression&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;variance of log-expression&quot;</span>)</span>
<span id="cb134-8"><a href="human-DLPFC-workflow.html#cb134-8" aria-hidden="true" tabindex="-1"></a><span class="fu">curve</span>(fit<span class="sc">$</span><span class="fu">trend</span>(x), <span class="at">col =</span> <span class="st">&quot;dodgerblue&quot;</span>, <span class="at">add =</span> <span class="cn">TRUE</span>, <span class="at">lwd =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="workflow_human_DLPFC_files/figure-html/feature_selection_HVGs-1.png" width="672" /></p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="human-DLPFC-workflow.html#cb135-1" aria-hidden="true" tabindex="-1"></a><span class="co"># select top HVGs</span></span>
<span id="cb135-2"><a href="human-DLPFC-workflow.html#cb135-2" aria-hidden="true" tabindex="-1"></a>top_hvgs <span class="ot">&lt;-</span> <span class="fu">getTopHVGs</span>(dec, <span class="at">prop =</span> <span class="fl">0.1</span>)</span>
<span id="cb135-3"><a href="human-DLPFC-workflow.html#cb135-3" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(top_hvgs)</span></code></pre></div>
<pre><code>## [1] 1448</code></pre>
</div>
<div id="dimensionality-reduction" class="section level2" number="16.7">
<h2><span class="header-section-number">16.7</span> Dimensionality reduction</h2>
<p>Run principal component analysis (PCA) to the set of top HVGs, and retain the top 50 principal components (PCs) for further downstream analyses. This is done both to reduce noise and to improve computational efficiency. We also run UMAP on the set of top 50 PCs and retain the top 2 UMAP components for visualization purposes.</p>
<p>We use the computationally efficient implementation of PCA available in <code>scater</code> <span class="citation">(<a href="mouse-coronal-workflow.html#ref-McCarthy2017-zd" role="doc-biblioref">McCarthy et al. 2017</a>)</span>, which uses randomization, and therefore requires setting a random seed for reproducibility.</p>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="human-DLPFC-workflow.html#cb137-1" aria-hidden="true" tabindex="-1"></a><span class="co"># compute PCA</span></span>
<span id="cb137-2"><a href="human-DLPFC-workflow.html#cb137-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb137-3"><a href="human-DLPFC-workflow.html#cb137-3" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(spe, <span class="at">subset_row =</span> top_hvgs)</span>
<span id="cb137-4"><a href="human-DLPFC-workflow.html#cb137-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-5"><a href="human-DLPFC-workflow.html#cb137-5" aria-hidden="true" tabindex="-1"></a><span class="fu">reducedDimNames</span>(spe)</span></code></pre></div>
<pre><code>## [1] &quot;PCA&quot;</code></pre>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="human-DLPFC-workflow.html#cb139-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">reducedDim</span>(spe, <span class="st">&quot;PCA&quot;</span>))</span></code></pre></div>
<pre><code>## [1] 3582   50</code></pre>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="human-DLPFC-workflow.html#cb141-1" aria-hidden="true" tabindex="-1"></a><span class="co"># compute UMAP on top 50 PCs</span></span>
<span id="cb141-2"><a href="human-DLPFC-workflow.html#cb141-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb141-3"><a href="human-DLPFC-workflow.html#cb141-3" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(spe, <span class="at">dimred =</span> <span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb141-4"><a href="human-DLPFC-workflow.html#cb141-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb141-5"><a href="human-DLPFC-workflow.html#cb141-5" aria-hidden="true" tabindex="-1"></a><span class="fu">reducedDimNames</span>(spe)</span></code></pre></div>
<pre><code>## [1] &quot;PCA&quot;  &quot;UMAP&quot;</code></pre>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="human-DLPFC-workflow.html#cb143-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">reducedDim</span>(spe, <span class="st">&quot;UMAP&quot;</span>))</span></code></pre></div>
<pre><code>## [1] 3582    2</code></pre>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="human-DLPFC-workflow.html#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="co"># update column names for plotting functions</span></span>
<span id="cb145-2"><a href="human-DLPFC-workflow.html#cb145-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(<span class="fu">reducedDim</span>(spe, <span class="st">&quot;UMAP&quot;</span>)) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;UMAP&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="clustering-1" class="section level2" number="16.8">
<h2><span class="header-section-number">16.8</span> Clustering</h2>
<p>Next, we perform clustering to define cell types. Here, we use molecular features (gene expression) only. Alternative approaches for spatial data that take into account spatial and other features will be described in <a href="clustering.html#clustering">Clustering</a>. We apply graph-based clustering using the Walktrap method implemented in <code>scran</code> <span class="citation">(<a href="mouse-coronal-workflow.html#ref-Lun2016-dn" role="doc-biblioref">Lun, McCarthy, and Marioni 2016</a>)</span>, applied to the top 50 PCs calculated on the set of top HVGs.</p>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="human-DLPFC-workflow.html#cb146-1" aria-hidden="true" tabindex="-1"></a><span class="co"># graph-based clustering</span></span>
<span id="cb146-2"><a href="human-DLPFC-workflow.html#cb146-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb146-3"><a href="human-DLPFC-workflow.html#cb146-3" aria-hidden="true" tabindex="-1"></a>k <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb146-4"><a href="human-DLPFC-workflow.html#cb146-4" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">buildSNNGraph</span>(spe, <span class="at">k =</span> k, <span class="at">use.dimred =</span> <span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb146-5"><a href="human-DLPFC-workflow.html#cb146-5" aria-hidden="true" tabindex="-1"></a>g_walk <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">cluster_walktrap</span>(g)</span>
<span id="cb146-6"><a href="human-DLPFC-workflow.html#cb146-6" aria-hidden="true" tabindex="-1"></a>clus <span class="ot">&lt;-</span> g_walk<span class="sc">$</span>membership</span>
<span id="cb146-7"><a href="human-DLPFC-workflow.html#cb146-7" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(clus)</span></code></pre></div>
<pre><code>## clus
##    1    2    3    4    5    6 
##  372  916  342 1083  349  520</code></pre>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="human-DLPFC-workflow.html#cb148-1" aria-hidden="true" tabindex="-1"></a><span class="co"># store cluster labels in column &#39;label&#39; in colData</span></span>
<span id="cb148-2"><a href="human-DLPFC-workflow.html#cb148-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colLabels</span>(spe) <span class="ot">&lt;-</span> <span class="fu">factor</span>(clus)</span></code></pre></div>
<p>Visualize the clusters by plotting in (i) spatial x-y coordinates on the tissue slide, and (ii) UMAP dimensions. We use plotting functions from the <a href="https://github.com/lmweber/ggpavis">ggspavis</a> package.</p>
<p>From the visualizations, we can see that the clustering reproduces the known biological structure (cortical layers), although not perfectly. The clusters are also separated in UMAP space, but again not perfectly.</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="human-DLPFC-workflow.html#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggspavis)</span>
<span id="cb149-2"><a href="human-DLPFC-workflow.html#cb149-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb149-3"><a href="human-DLPFC-workflow.html#cb149-3" aria-hidden="true" tabindex="-1"></a><span class="co"># plot clusters in spatial x-y coordinates</span></span>
<span id="cb149-4"><a href="human-DLPFC-workflow.html#cb149-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSpots</span>(spe, <span class="at">discrete =</span> <span class="st">&quot;label&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;libd_layer_colors&quot;</span>)</span></code></pre></div>
<p><img src="workflow_human_DLPFC_files/figure-html/clustering_plots-1.png" width="528" /></p>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="human-DLPFC-workflow.html#cb150-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot ground truth labels in spatial x-y coordinates</span></span>
<span id="cb150-2"><a href="human-DLPFC-workflow.html#cb150-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSpots</span>(spe, <span class="at">discrete =</span> <span class="st">&quot;ground_truth&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;libd_layer_colors&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: Removed 14 rows containing missing values (geom_point).</code></pre>
<p><img src="workflow_human_DLPFC_files/figure-html/clustering_plots-2.png" width="528" /></p>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="human-DLPFC-workflow.html#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot clusters in UMAP reduced dimensions</span></span>
<span id="cb152-2"><a href="human-DLPFC-workflow.html#cb152-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDimRed</span>(spe, <span class="at">type =</span> <span class="st">&quot;UMAP&quot;</span>, <span class="at">discrete =</span> <span class="st">&quot;label&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;libd_layer_colors&quot;</span>)</span></code></pre></div>
<p><img src="workflow_human_DLPFC_files/figure-html/clustering_plots-3.png" width="528" /></p>
</div>
<div id="marker-genes-1" class="section level2" number="16.9">
<h2><span class="header-section-number">16.9</span> Marker genes</h2>
<p>Identify marker genes by testing for differential gene expression between clusters. We use the <code>findMarkers</code> implementation in <code>scran</code> <span class="citation">(<a href="mouse-coronal-workflow.html#ref-Lun2016-dn" role="doc-biblioref">Lun, McCarthy, and Marioni 2016</a>)</span>, using a binomial test, which tests for genes that differ in the proportion expressed vs. not expressed between clusters. This is a more stringent test than the default t-tests, and tends to select genes that are easier to interpret and validate experimentally.</p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="human-DLPFC-workflow.html#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="co"># set gene names as row names for visualization purposes</span></span>
<span id="cb153-2"><a href="human-DLPFC-workflow.html#cb153-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(spe) <span class="ot">&lt;-</span> <span class="fu">rowData</span>(spe)<span class="sc">$</span>gene_name</span>
<span id="cb153-3"><a href="human-DLPFC-workflow.html#cb153-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb153-4"><a href="human-DLPFC-workflow.html#cb153-4" aria-hidden="true" tabindex="-1"></a><span class="co"># test for marker genes</span></span>
<span id="cb153-5"><a href="human-DLPFC-workflow.html#cb153-5" aria-hidden="true" tabindex="-1"></a>markers <span class="ot">&lt;-</span> <span class="fu">findMarkers</span>(spe, <span class="at">test =</span> <span class="st">&quot;binom&quot;</span>, <span class="at">direction =</span> <span class="st">&quot;up&quot;</span>)</span>
<span id="cb153-6"><a href="human-DLPFC-workflow.html#cb153-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb153-7"><a href="human-DLPFC-workflow.html#cb153-7" aria-hidden="true" tabindex="-1"></a><span class="co"># returns a list with one DataFrame per cluster</span></span>
<span id="cb153-8"><a href="human-DLPFC-workflow.html#cb153-8" aria-hidden="true" tabindex="-1"></a>markers</span></code></pre></div>
<pre><code>## List of length 6
## names(6): 1 2 3 4 5 6</code></pre>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="human-DLPFC-workflow.html#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb155-2"><a href="human-DLPFC-workflow.html#cb155-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-3"><a href="human-DLPFC-workflow.html#cb155-3" aria-hidden="true" tabindex="-1"></a><span class="co"># plot log-fold changes for one cluster over all other clusters</span></span>
<span id="cb155-4"><a href="human-DLPFC-workflow.html#cb155-4" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting cluster 1</span></span>
<span id="cb155-5"><a href="human-DLPFC-workflow.html#cb155-5" aria-hidden="true" tabindex="-1"></a>interesting <span class="ot">&lt;-</span> markers[[<span class="dv">1</span>]]</span>
<span id="cb155-6"><a href="human-DLPFC-workflow.html#cb155-6" aria-hidden="true" tabindex="-1"></a>best_set <span class="ot">&lt;-</span> interesting[interesting<span class="sc">$</span>Top <span class="sc">&lt;=</span> <span class="dv">5</span>, ]</span>
<span id="cb155-7"><a href="human-DLPFC-workflow.html#cb155-7" aria-hidden="true" tabindex="-1"></a>logFCs <span class="ot">&lt;-</span> <span class="fu">getMarkerEffects</span>(best_set)</span>
<span id="cb155-8"><a href="human-DLPFC-workflow.html#cb155-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-9"><a href="human-DLPFC-workflow.html#cb155-9" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(logFCs, <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">5</span>, <span class="at">length.out =</span> <span class="dv">101</span>))</span></code></pre></div>
<p><img src="workflow_human_DLPFC_files/figure-html/marker_genes_heatmap-1.png" width="480" /></p>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="human-DLPFC-workflow.html#cb156-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot log-transformed normalized expression of top genes for one cluster</span></span>
<span id="cb156-2"><a href="human-DLPFC-workflow.html#cb156-2" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> <span class="fu">head</span>(<span class="fu">rownames</span>(interesting))</span>
<span id="cb156-3"><a href="human-DLPFC-workflow.html#cb156-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-4"><a href="human-DLPFC-workflow.html#cb156-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotExpression</span>(spe, <span class="at">x =</span> <span class="st">&quot;label&quot;</span>, <span class="at">features =</span> top_genes)</span></code></pre></div>
<p><img src="workflow_human_DLPFC_files/figure-html/marker_genes_expression-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="workflows.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="mouse-coronal-workflow.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/workflow_human_DLPFC.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OSTA-book.pdf", "OSTA-book.epub"],
"toc": {
"collapse": "subsection"
},
"lib_dir": "book_assets"
});
});
</script>

</body>

</html>
