<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 18 Visium mouse coronal workflow | Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor</title>
  <meta name="description" content="Online textbook on ‘Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 18 Visium mouse coronal workflow | Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Online textbook on ‘Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor’" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 18 Visium mouse coronal workflow | Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor" />
  
  <meta name="twitter:description" content="Online textbook on ‘Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor’" />
  



<meta name="date" content="2021-03-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="visium-human-dlpfc-workflow.html"/>
<link rel="next" href="visium-spatiallibd-workflow.html"/>
<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">OSTA</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#contents-of-this-book"><i class="fa fa-check"></i><b>1.1</b> Contents of this book</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#who-this-book-is-for"><i class="fa fa-check"></i><b>1.2</b> Who this book is for</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#bioconductor"><i class="fa fa-check"></i><b>1.3</b> Bioconductor</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="spatially-resolved-transcriptomics.html"><a href="spatially-resolved-transcriptomics.html"><i class="fa fa-check"></i><b>2</b> Spatially resolved transcriptomics</a>
<ul>
<li class="chapter" data-level="2.1" data-path="spatially-resolved-transcriptomics.html"><a href="spatially-resolved-transcriptomics.html#x-genomics-visium"><i class="fa fa-check"></i><b>2.1</b> 10x Genomics Visium</a></li>
<li class="chapter" data-level="2.2" data-path="spatially-resolved-transcriptomics.html"><a href="spatially-resolved-transcriptomics.html#cartana"><i class="fa fa-check"></i><b>2.2</b> Cartana</a></li>
<li class="chapter" data-level="2.3" data-path="spatially-resolved-transcriptomics.html"><a href="spatially-resolved-transcriptomics.html#slide-seqv2"><i class="fa fa-check"></i><b>2.3</b> Slide-seqV2</a></li>
<li class="chapter" data-level="2.4" data-path="spatially-resolved-transcriptomics.html"><a href="spatially-resolved-transcriptomics.html#seqfish"><i class="fa fa-check"></i><b>2.4</b> seqFISH</a></li>
<li class="chapter" data-level="2.5" data-path="spatially-resolved-transcriptomics.html"><a href="spatially-resolved-transcriptomics.html#merfish"><i class="fa fa-check"></i><b>2.5</b> MERFISH</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spatialexperiment.html"><a href="spatialexperiment.html"><i class="fa fa-check"></i><b>3</b> SpatialExperiment</a>
<ul>
<li class="chapter" data-level="3.1" data-path="spatialexperiment.html"><a href="spatialexperiment.html#spatialexperiment-class"><i class="fa fa-check"></i><b>3.1</b> SpatialExperiment class</a></li>
</ul></li>
<li class="part"><span><b>II Preprocessing steps</b></span></li>
<li class="chapter" data-level="4" data-path="preprocessing-steps.html"><a href="preprocessing-steps.html"><i class="fa fa-check"></i><b>4</b> Preprocessing steps</a></li>
<li class="chapter" data-level="5" data-path="image-segmentation-visium.html"><a href="image-segmentation-visium.html"><i class="fa fa-check"></i><b>5</b> Image segmentation (Visium)</a>
<ul>
<li class="chapter" data-level="5.1" data-path="image-segmentation-visium.html"><a href="image-segmentation-visium.html#overview"><i class="fa fa-check"></i><b>5.1</b> Overview</a></li>
<li class="chapter" data-level="5.2" data-path="image-segmentation-visium.html"><a href="image-segmentation-visium.html#split-histology-images"><i class="fa fa-check"></i><b>5.2</b> Split histology images</a></li>
<li class="chapter" data-level="5.3" data-path="image-segmentation-visium.html"><a href="image-segmentation-visium.html#file-size-issues"><i class="fa fa-check"></i><b>5.3</b> File size issues</a></li>
<li class="chapter" data-level="5.4" data-path="image-segmentation-visium.html"><a href="image-segmentation-visium.html#segment-cell-nuclei"><i class="fa fa-check"></i><b>5.4</b> Segment cell nuclei</a></li>
<li class="chapter" data-level="5.5" data-path="image-segmentation-visium.html"><a href="image-segmentation-visium.html#identify-number-of-cells-per-spot"><i class="fa fa-check"></i><b>5.5</b> Identify number of cells per spot</a></li>
<li class="chapter" data-level="5.6" data-path="image-segmentation-visium.html"><a href="image-segmentation-visium.html#output-files"><i class="fa fa-check"></i><b>5.6</b> Output files</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="loupe-browser-visium.html"><a href="loupe-browser-visium.html"><i class="fa fa-check"></i><b>6</b> Loupe Browser (Visium)</a>
<ul>
<li class="chapter" data-level="6.1" data-path="loupe-browser-visium.html"><a href="loupe-browser-visium.html#overview-1"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="loupe-browser-visium.html"><a href="loupe-browser-visium.html#automatic-alignment-of-images"><i class="fa fa-check"></i><b>6.2</b> Automatic alignment of images</a></li>
<li class="chapter" data-level="6.3" data-path="loupe-browser-visium.html"><a href="loupe-browser-visium.html#manual-alignment-of-images"><i class="fa fa-check"></i><b>6.3</b> Manual alignment of images</a></li>
<li class="chapter" data-level="6.4" data-path="loupe-browser-visium.html"><a href="loupe-browser-visium.html#output-files-for-space-ranger"><i class="fa fa-check"></i><b>6.4</b> Output files for Space Ranger</a></li>
<li class="chapter" data-level="6.5" data-path="loupe-browser-visium.html"><a href="loupe-browser-visium.html#downstream-analyses-in-loupe-browser"><i class="fa fa-check"></i><b>6.5</b> Downstream analyses in Loupe Browser</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="space-ranger-visium.html"><a href="space-ranger-visium.html"><i class="fa fa-check"></i><b>7</b> Space Ranger (Visium)</a>
<ul>
<li class="chapter" data-level="7.1" data-path="space-ranger-visium.html"><a href="space-ranger-visium.html#overview-2"><i class="fa fa-check"></i><b>7.1</b> Overview</a></li>
<li class="chapter" data-level="7.2" data-path="space-ranger-visium.html"><a href="space-ranger-visium.html#installation"><i class="fa fa-check"></i><b>7.2</b> Installation</a></li>
<li class="chapter" data-level="7.3" data-path="space-ranger-visium.html"><a href="space-ranger-visium.html#memory-issues"><i class="fa fa-check"></i><b>7.3</b> Memory issues</a></li>
<li class="chapter" data-level="7.4" data-path="space-ranger-visium.html"><a href="space-ranger-visium.html#run-spaceranger-count"><i class="fa fa-check"></i><b>7.4</b> Run spaceranger count</a></li>
<li class="chapter" data-level="7.5" data-path="space-ranger-visium.html"><a href="space-ranger-visium.html#output-files-1"><i class="fa fa-check"></i><b>7.5</b> Output files</a></li>
<li class="chapter" data-level="7.6" data-path="space-ranger-visium.html"><a href="space-ranger-visium.html#web-summary-.html-file"><i class="fa fa-check"></i><b>7.6</b> Web summary .html file</a></li>
<li class="chapter" data-level="7.7" data-path="space-ranger-visium.html"><a href="space-ranger-visium.html#import-outputs-into-r"><i class="fa fa-check"></i><b>7.7</b> Import outputs into R</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="other-platforms.html"><a href="other-platforms.html"><i class="fa fa-check"></i><b>8</b> Other platforms</a></li>
<li class="part"><span><b>III Analysis steps</b></span></li>
<li class="chapter" data-level="9" data-path="analysis-steps.html"><a href="analysis-steps.html"><i class="fa fa-check"></i><b>9</b> Analysis steps</a>
<ul>
<li class="chapter" data-level="9.1" data-path="analysis-steps.html"><a href="analysis-steps.html#load-data"><i class="fa fa-check"></i><b>9.1</b> Load data</a></li>
<li class="chapter" data-level="9.2" data-path="analysis-steps.html"><a href="analysis-steps.html#spatialexperiment-object-structure"><i class="fa fa-check"></i><b>9.2</b> SpatialExperiment object structure</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>10</b> Quality control</a>
<ul>
<li class="chapter" data-level="10.1" data-path="quality-control.html"><a href="quality-control.html#background"><i class="fa fa-check"></i><b>10.1</b> Background</a></li>
<li class="chapter" data-level="10.2" data-path="quality-control.html"><a href="quality-control.html#load-data-1"><i class="fa fa-check"></i><b>10.2</b> Load data</a></li>
<li class="chapter" data-level="10.3" data-path="quality-control.html"><a href="quality-control.html#plot-data"><i class="fa fa-check"></i><b>10.3</b> Plot data</a></li>
<li class="chapter" data-level="10.4" data-path="quality-control.html"><a href="quality-control.html#calculate-qc-metrics"><i class="fa fa-check"></i><b>10.4</b> Calculate QC metrics</a></li>
<li class="chapter" data-level="10.5" data-path="quality-control.html"><a href="quality-control.html#selecting-thresholds"><i class="fa fa-check"></i><b>10.5</b> Selecting thresholds</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="quality-control.html"><a href="quality-control.html#library-size"><i class="fa fa-check"></i><b>10.5.1</b> Library size</a></li>
<li class="chapter" data-level="10.5.2" data-path="quality-control.html"><a href="quality-control.html#number-of-expressed-features"><i class="fa fa-check"></i><b>10.5.2</b> Number of expressed features</a></li>
<li class="chapter" data-level="10.5.3" data-path="quality-control.html"><a href="quality-control.html#proportion-of-mitochondrial-reads"><i class="fa fa-check"></i><b>10.5.3</b> Proportion of mitochondrial reads</a></li>
<li class="chapter" data-level="10.5.4" data-path="quality-control.html"><a href="quality-control.html#number-of-cells-per-spot"><i class="fa fa-check"></i><b>10.5.4</b> Number of cells per spot</a></li>
<li class="chapter" data-level="10.5.5" data-path="quality-control.html"><a href="quality-control.html#remove-low-quality-spots"><i class="fa fa-check"></i><b>10.5.5</b> Remove low-quality spots</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="quality-control.html"><a href="quality-control.html#zero-cell-and-single-cell-spots"><i class="fa fa-check"></i><b>10.6</b> Zero-cell and single-cell spots</a></li>
<li class="chapter" data-level="10.7" data-path="quality-control.html"><a href="quality-control.html#quality-control-at-gene-level"><i class="fa fa-check"></i><b>10.7</b> Quality control at gene level</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="normalization.html"><a href="normalization.html"><i class="fa fa-check"></i><b>11</b> Normalization</a>
<ul>
<li class="chapter" data-level="11.1" data-path="normalization.html"><a href="normalization.html#background-1"><i class="fa fa-check"></i><b>11.1</b> Background</a></li>
<li class="chapter" data-level="11.2" data-path="normalization.html"><a href="normalization.html#previous-steps"><i class="fa fa-check"></i><b>11.2</b> Previous steps</a></li>
<li class="chapter" data-level="11.3" data-path="normalization.html"><a href="normalization.html#log-transformed-normalized-counts"><i class="fa fa-check"></i><b>11.3</b> Log-transformed normalized counts</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="feature-selection.html"><a href="feature-selection.html"><i class="fa fa-check"></i><b>12</b> Feature selection</a>
<ul>
<li class="chapter" data-level="12.1" data-path="feature-selection.html"><a href="feature-selection.html#background-2"><i class="fa fa-check"></i><b>12.1</b> Background</a></li>
<li class="chapter" data-level="12.2" data-path="feature-selection.html"><a href="feature-selection.html#previous-steps-1"><i class="fa fa-check"></i><b>12.2</b> Previous steps</a></li>
<li class="chapter" data-level="12.3" data-path="feature-selection.html"><a href="feature-selection.html#highly-variable-genes-hvgs"><i class="fa fa-check"></i><b>12.3</b> Highly variable genes (HVGs)</a></li>
<li class="chapter" data-level="12.4" data-path="feature-selection.html"><a href="feature-selection.html#spatially-variable-genes-svgs"><i class="fa fa-check"></i><b>12.4</b> Spatially variable genes (SVGs)</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="feature-selection.html"><a href="feature-selection.html#spatialde"><i class="fa fa-check"></i><b>12.4.1</b> SpatialDE</a></li>
<li class="chapter" data-level="12.4.2" data-path="feature-selection.html"><a href="feature-selection.html#spark"><i class="fa fa-check"></i><b>12.4.2</b> SPARK</a></li>
<li class="chapter" data-level="12.4.3" data-path="feature-selection.html"><a href="feature-selection.html#morans-i-statistic"><i class="fa fa-check"></i><b>12.4.3</b> Moran’s I statistic</a></li>
<li class="chapter" data-level="12.4.4" data-path="feature-selection.html"><a href="feature-selection.html#further-methods"><i class="fa fa-check"></i><b>12.4.4</b> Further methods</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="dimensionality-reduction.html"><a href="dimensionality-reduction.html"><i class="fa fa-check"></i><b>13</b> Dimensionality reduction</a>
<ul>
<li class="chapter" data-level="13.1" data-path="dimensionality-reduction.html"><a href="dimensionality-reduction.html#background-3"><i class="fa fa-check"></i><b>13.1</b> Background</a></li>
<li class="chapter" data-level="13.2" data-path="dimensionality-reduction.html"><a href="dimensionality-reduction.html#previous-steps-2"><i class="fa fa-check"></i><b>13.2</b> Previous steps</a></li>
<li class="chapter" data-level="13.3" data-path="dimensionality-reduction.html"><a href="dimensionality-reduction.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>13.3</b> Principal component analysis (PCA)</a></li>
<li class="chapter" data-level="13.4" data-path="dimensionality-reduction.html"><a href="dimensionality-reduction.html#uniform-manifold-approximation-and-projection-umap"><i class="fa fa-check"></i><b>13.4</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="13.5" data-path="dimensionality-reduction.html"><a href="dimensionality-reduction.html#visualizations"><i class="fa fa-check"></i><b>13.5</b> Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="clustering.html"><a href="clustering.html"><i class="fa fa-check"></i><b>14</b> Clustering</a>
<ul>
<li class="chapter" data-level="14.1" data-path="clustering.html"><a href="clustering.html#overview-3"><i class="fa fa-check"></i><b>14.1</b> Overview</a></li>
<li class="chapter" data-level="14.2" data-path="clustering.html"><a href="clustering.html#previous-steps-3"><i class="fa fa-check"></i><b>14.2</b> Previous steps</a></li>
<li class="chapter" data-level="14.3" data-path="clustering.html"><a href="clustering.html#clustering-on-hvgs"><i class="fa fa-check"></i><b>14.3</b> Clustering on HVGs</a></li>
<li class="chapter" data-level="14.4" data-path="clustering.html"><a href="clustering.html#clustering-on-hvgs-and-spatial-dimensions"><i class="fa fa-check"></i><b>14.4</b> Clustering on HVGs and spatial dimensions</a></li>
<li class="chapter" data-level="14.5" data-path="clustering.html"><a href="clustering.html#bayesspace"><i class="fa fa-check"></i><b>14.5</b> BayesSpace</a></li>
<li class="chapter" data-level="14.6" data-path="clustering.html"><a href="clustering.html#further-methods-1"><i class="fa fa-check"></i><b>14.6</b> Further methods</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="marker-genes.html"><a href="marker-genes.html"><i class="fa fa-check"></i><b>15</b> Marker genes</a>
<ul>
<li class="chapter" data-level="15.1" data-path="marker-genes.html"><a href="marker-genes.html#background-4"><i class="fa fa-check"></i><b>15.1</b> Background</a></li>
<li class="chapter" data-level="15.2" data-path="marker-genes.html"><a href="marker-genes.html#previous-steps-4"><i class="fa fa-check"></i><b>15.2</b> Previous steps</a></li>
<li class="chapter" data-level="15.3" data-path="marker-genes.html"><a href="marker-genes.html#marker-genes-1"><i class="fa fa-check"></i><b>15.3</b> Marker genes</a></li>
</ul></li>
<li class="part"><span><b>IV Workflows</b></span></li>
<li class="chapter" data-level="16" data-path="workflows.html"><a href="workflows.html"><i class="fa fa-check"></i><b>16</b> Workflows</a></li>
<li class="chapter" data-level="17" data-path="visium-human-dlpfc-workflow.html"><a href="visium-human-dlpfc-workflow.html"><i class="fa fa-check"></i><b>17</b> Visium human DLPFC workflow</a>
<ul>
<li class="chapter" data-level="17.1" data-path="visium-human-dlpfc-workflow.html"><a href="visium-human-dlpfc-workflow.html#description-of-dataset"><i class="fa fa-check"></i><b>17.1</b> Description of dataset</a></li>
<li class="chapter" data-level="17.2" data-path="visium-human-dlpfc-workflow.html"><a href="visium-human-dlpfc-workflow.html#load-data-2"><i class="fa fa-check"></i><b>17.2</b> Load data</a></li>
<li class="chapter" data-level="17.3" data-path="visium-human-dlpfc-workflow.html"><a href="visium-human-dlpfc-workflow.html#plot-data-1"><i class="fa fa-check"></i><b>17.3</b> Plot data</a></li>
<li class="chapter" data-level="17.4" data-path="visium-human-dlpfc-workflow.html"><a href="visium-human-dlpfc-workflow.html#quality-control-qc"><i class="fa fa-check"></i><b>17.4</b> Quality control (QC)</a></li>
<li class="chapter" data-level="17.5" data-path="visium-human-dlpfc-workflow.html"><a href="visium-human-dlpfc-workflow.html#normalization-1"><i class="fa fa-check"></i><b>17.5</b> Normalization</a></li>
<li class="chapter" data-level="17.6" data-path="visium-human-dlpfc-workflow.html"><a href="visium-human-dlpfc-workflow.html#feature-selection-1"><i class="fa fa-check"></i><b>17.6</b> Feature selection</a></li>
<li class="chapter" data-level="17.7" data-path="visium-human-dlpfc-workflow.html"><a href="visium-human-dlpfc-workflow.html#dimensionality-reduction-1"><i class="fa fa-check"></i><b>17.7</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="17.8" data-path="visium-human-dlpfc-workflow.html"><a href="visium-human-dlpfc-workflow.html#clustering-1"><i class="fa fa-check"></i><b>17.8</b> Clustering</a></li>
<li class="chapter" data-level="17.9" data-path="visium-human-dlpfc-workflow.html"><a href="visium-human-dlpfc-workflow.html#marker-genes-2"><i class="fa fa-check"></i><b>17.9</b> Marker genes</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="visium-mouse-coronal-workflow.html"><a href="visium-mouse-coronal-workflow.html"><i class="fa fa-check"></i><b>18</b> Visium mouse coronal workflow</a>
<ul>
<li class="chapter" data-level="18.1" data-path="visium-mouse-coronal-workflow.html"><a href="visium-mouse-coronal-workflow.html#description-of-dataset-1"><i class="fa fa-check"></i><b>18.1</b> Description of dataset</a></li>
<li class="chapter" data-level="18.2" data-path="visium-mouse-coronal-workflow.html"><a href="visium-mouse-coronal-workflow.html#load-data-3"><i class="fa fa-check"></i><b>18.2</b> Load data</a></li>
<li class="chapter" data-level="18.3" data-path="visium-mouse-coronal-workflow.html"><a href="visium-mouse-coronal-workflow.html#plot-data-2"><i class="fa fa-check"></i><b>18.3</b> Plot data</a></li>
<li class="chapter" data-level="18.4" data-path="visium-mouse-coronal-workflow.html"><a href="visium-mouse-coronal-workflow.html#quality-control-qc-1"><i class="fa fa-check"></i><b>18.4</b> Quality control (QC)</a></li>
<li class="chapter" data-level="18.5" data-path="visium-mouse-coronal-workflow.html"><a href="visium-mouse-coronal-workflow.html#normalization-2"><i class="fa fa-check"></i><b>18.5</b> Normalization</a></li>
<li class="chapter" data-level="18.6" data-path="visium-mouse-coronal-workflow.html"><a href="visium-mouse-coronal-workflow.html#feature-selection-2"><i class="fa fa-check"></i><b>18.6</b> Feature selection</a></li>
<li class="chapter" data-level="18.7" data-path="visium-mouse-coronal-workflow.html"><a href="visium-mouse-coronal-workflow.html#dimensionality-reduction-2"><i class="fa fa-check"></i><b>18.7</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="18.8" data-path="visium-mouse-coronal-workflow.html"><a href="visium-mouse-coronal-workflow.html#clustering-2"><i class="fa fa-check"></i><b>18.8</b> Clustering</a></li>
<li class="chapter" data-level="18.9" data-path="visium-mouse-coronal-workflow.html"><a href="visium-mouse-coronal-workflow.html#marker-genes-3"><i class="fa fa-check"></i><b>18.9</b> Marker genes</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="visium-spatiallibd-workflow.html"><a href="visium-spatiallibd-workflow.html"><i class="fa fa-check"></i><b>19</b> Visium spatialLIBD workflow</a></li>
<li class="chapter" data-level="20" data-path="seqfish-workflow.html"><a href="seqfish-workflow.html"><i class="fa fa-check"></i><b>20</b> seqFISH workflow</a></li>
<li class="chapter" data-level="21" data-path="cartana-workflow.html"><a href="cartana-workflow.html"><i class="fa fa-check"></i><b>21</b> Cartana workflow</a></li>
<li class="part"><span><b>V Appendix</b></span></li>
<li class="chapter" data-level="22" data-path="contributors.html"><a href="contributors.html"><i class="fa fa-check"></i><b>22</b> Contributors</a></li>
<li class="chapter" data-level="23" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>23</b> References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="visium-mouse-coronal-workflow" class="section level1" number="18">
<h1><span class="header-section-number">Chapter 18</span> Visium mouse coronal workflow</h1>
<p>This workflow analyzes a mouse coronal brain section dataset from the 10x Genomics Visium platform. This dataset was generated by 10x Genomics, and the raw data files are publicly available from the <a href="https://support.10xgenomics.com/spatial-gene-expression/datasets/1.1.0/V1_Adult_Mouse_Brain">10x Genomics website</a>.</p>
<div id="description-of-dataset-1" class="section level2" number="18.1">
<h2><span class="header-section-number">18.1</span> Description of dataset</h2>
<p>This dataset measures transcriptome-wide gene expression on a Visium slide spanning one hemisphere of a mouse coronal brain section. For experimental details, see the <a href="https://support.10xgenomics.com/spatial-gene-expression/datasets/1.1.0/V1_Adult_Mouse_Brain">10x Genomics website</a>.</p>
<p>Due to the small size of the mouse brain and the dimensions of the Visium slide (6.5mm x 6.5mm), the measurements span an entire brain hemisphere. Therefore, we can use this dataset to compare gene expression profiles between major anatomical regions of the mouse brain. Due to the small size of cells in the mouse brain, each spot can contain up to 50 cells. In this dataset, we do not know the exact number of cells per spot.</p>
</div>
<div id="load-data-3" class="section level2" number="18.2">
<h2><span class="header-section-number">18.2</span> Load data</h2>
<p>The dataset is available in <code>SpatialExperiment</code> format from the <a href="https://github.com/lmweber/STexampleData">STexampleData</a> package.</p>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="visium-mouse-coronal-workflow.html#cb167-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SpatialExperiment)</span>
<span id="cb167-2"><a href="visium-mouse-coronal-workflow.html#cb167-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(STexampleData)</span>
<span id="cb167-3"><a href="visium-mouse-coronal-workflow.html#cb167-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb167-4"><a href="visium-mouse-coronal-workflow.html#cb167-4" aria-hidden="true" tabindex="-1"></a><span class="co"># load object</span></span>
<span id="cb167-5"><a href="visium-mouse-coronal-workflow.html#cb167-5" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> <span class="fu">load_data</span>(<span class="st">&quot;Visium_mouseCoronal&quot;</span>)</span>
<span id="cb167-6"><a href="visium-mouse-coronal-workflow.html#cb167-6" aria-hidden="true" tabindex="-1"></a>spe</span></code></pre></div>
<pre><code>## class: SpatialExperiment 
## dim: 32285 4992 
## metadata(0):
## assays(1): counts
## rownames(32285): ENSMUSG00000051951 ENSMUSG00000089699 ...
##   ENSMUSG00000095019 ENSMUSG00000095041
## rowData names(3): gene_id gene_name feature_type
## colnames(4992): AAACAACGAATAGTTC-1 AAACAAGTATCTCCCA-1 ...
##   TTGTTTGTATTACACG-1 TTGTTTGTGTAAATTC-1
## colData names(6): barcode_id sample_id ... pxl_col_in_fullres
##   pxl_row_in_fullres
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## spatialData names(4) : barcode_id in_tissue x y
## imgData names(4): sample_id image_id data scaleFactor</code></pre>
</div>
<div id="plot-data-2" class="section level2" number="18.3">
<h2><span class="header-section-number">18.3</span> Plot data</h2>
<p>As an initial check, plot the spatial coordinates (spots) in x-y dimensions on the tissue slide. This confirms that the object has loaded correctly, and the orientation matches the <a href="https://support.10xgenomics.com/spatial-gene-expression/datasets/1.1.0/V1_Adult_Mouse_Brain">10x Genomics website</a>.</p>
<p>We use visualization functions from the <a href="https://github.com/lmweber/ggpavis">ggspavis</a> package to generate plots.</p>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="visium-mouse-coronal-workflow.html#cb169-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggspavis)</span>
<span id="cb169-2"><a href="visium-mouse-coronal-workflow.html#cb169-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb169-3"><a href="visium-mouse-coronal-workflow.html#cb169-3" aria-hidden="true" tabindex="-1"></a><span class="co"># plot spatial coordinates (spots)</span></span>
<span id="cb169-4"><a href="visium-mouse-coronal-workflow.html#cb169-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSpots</span>(spe)</span></code></pre></div>
<p><img src="workflow_Visium_mouseCoronal_files/figure-html/plot_data-1.png" width="672" /></p>
</div>
<div id="quality-control-qc-1" class="section level2" number="18.4">
<h2><span class="header-section-number">18.4</span> Quality control (QC)</h2>
<p>Subset object to keep only spots over tissue.</p>
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb170-1"><a href="visium-mouse-coronal-workflow.html#cb170-1" aria-hidden="true" tabindex="-1"></a><span class="co"># subset to keep only spots over tissue</span></span>
<span id="cb170-2"><a href="visium-mouse-coronal-workflow.html#cb170-2" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> spe[, <span class="fu">spatialData</span>(spe)<span class="sc">$</span>in_tissue <span class="sc">==</span> <span class="dv">1</span>]</span>
<span id="cb170-3"><a href="visium-mouse-coronal-workflow.html#cb170-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(spe)</span></code></pre></div>
<pre><code>## [1] 32285  2702</code></pre>
<p>Calculate spot-level QC metrics using the <code>scater</code> package <span class="citation">(<a href="references.html#ref-McCarthy2017" role="doc-biblioref">McCarthy et al. 2017</a>)</span>, and store the QC metrics in <code>colData</code>.</p>
<div class="sourceCode" id="cb172"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb172-1"><a href="visium-mouse-coronal-workflow.html#cb172-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb172-2"><a href="visium-mouse-coronal-workflow.html#cb172-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb172-3"><a href="visium-mouse-coronal-workflow.html#cb172-3" aria-hidden="true" tabindex="-1"></a><span class="co"># identify mitochondrial genes</span></span>
<span id="cb172-4"><a href="visium-mouse-coronal-workflow.html#cb172-4" aria-hidden="true" tabindex="-1"></a>is_mito <span class="ot">&lt;-</span> <span class="fu">grepl</span>(<span class="st">&quot;(^MT-)|(^mt-)&quot;</span>, <span class="fu">rowData</span>(spe)<span class="sc">$</span>gene_name)</span>
<span id="cb172-5"><a href="visium-mouse-coronal-workflow.html#cb172-5" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(is_mito)</span></code></pre></div>
<pre><code>## is_mito
## FALSE  TRUE 
## 32272    13</code></pre>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="visium-mouse-coronal-workflow.html#cb174-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(spe)<span class="sc">$</span>gene_name[is_mito]</span></code></pre></div>
<pre><code>##  [1] &quot;mt-Nd1&quot;  &quot;mt-Nd2&quot;  &quot;mt-Co1&quot;  &quot;mt-Co2&quot;  &quot;mt-Atp8&quot; &quot;mt-Atp6&quot; &quot;mt-Co3&quot; 
##  [8] &quot;mt-Nd3&quot;  &quot;mt-Nd4l&quot; &quot;mt-Nd4&quot;  &quot;mt-Nd5&quot;  &quot;mt-Nd6&quot;  &quot;mt-Cytb&quot;</code></pre>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="visium-mouse-coronal-workflow.html#cb176-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate per-spot QC metrics and store in colData</span></span>
<span id="cb176-2"><a href="visium-mouse-coronal-workflow.html#cb176-2" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> <span class="fu">addPerCellQC</span>(spe, <span class="at">subsets =</span> <span class="fu">list</span>(<span class="at">mito =</span> is_mito))</span>
<span id="cb176-3"><a href="visium-mouse-coronal-workflow.html#cb176-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">colData</span>(spe), <span class="dv">3</span>)</span></code></pre></div>
<pre><code>## DataFrame with 3 rows and 12 columns
##                            barcode_id   sample_id array_row array_col
##                           &lt;character&gt; &lt;character&gt; &lt;integer&gt; &lt;integer&gt;
## AAACAAGTATCTCCCA-1 AAACAAGTATCTCCCA-1   sample_01        50       102
## AAACAATCTACTAGCA-1 AAACAATCTACTAGCA-1   sample_01         3        43
## AAACACCAATAACTGC-1 AAACACCAATAACTGC-1   sample_01        59        19
##                    pxl_col_in_fullres pxl_row_in_fullres       sum  detected
##                             &lt;integer&gt;          &lt;integer&gt; &lt;numeric&gt; &lt;numeric&gt;
## AAACAAGTATCTCCCA-1               7237               8229     20935      5230
## AAACAATCTACTAGCA-1               1611               4169     14789      3646
## AAACACCAATAACTGC-1               8315               2518     34646      6272
##                    subsets_mito_sum subsets_mito_detected subsets_mito_percent
##                           &lt;numeric&gt;             &lt;numeric&gt;            &lt;numeric&gt;
## AAACAAGTATCTCCCA-1             4036                    13              19.2787
## AAACAATCTACTAGCA-1             3419                    13              23.1185
## AAACACCAATAACTGC-1             5068                    13              14.6280
##                        total
##                    &lt;numeric&gt;
## AAACAAGTATCTCCCA-1     20935
## AAACAATCTACTAGCA-1     14789
## AAACACCAATAACTGC-1     34646</code></pre>
<p>Select filtering thresholds for the QC metrics by examining distributions using histograms.</p>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="visium-mouse-coronal-workflow.html#cb178-1" aria-hidden="true" tabindex="-1"></a><span class="co"># histograms of QC metrics</span></span>
<span id="cb178-2"><a href="visium-mouse-coronal-workflow.html#cb178-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>))</span>
<span id="cb178-3"><a href="visium-mouse-coronal-workflow.html#cb178-3" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">colData</span>(spe)<span class="sc">$</span>sum, <span class="at">xlab =</span> <span class="st">&quot;sum&quot;</span>, <span class="at">main =</span> <span class="st">&quot;UMIs per spot&quot;</span>)</span>
<span id="cb178-4"><a href="visium-mouse-coronal-workflow.html#cb178-4" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">colData</span>(spe)<span class="sc">$</span>detected, <span class="at">xlab =</span> <span class="st">&quot;detected&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Genes per spot&quot;</span>)</span>
<span id="cb178-5"><a href="visium-mouse-coronal-workflow.html#cb178-5" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">colData</span>(spe)<span class="sc">$</span>subsets_mito_percent, <span class="at">xlab =</span> <span class="st">&quot;percent mitochondrial&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Percent mito UMIs&quot;</span>)</span></code></pre></div>
<p><img src="workflow_Visium_mouseCoronal_files/figure-html/QC_thresholds-1.png" width="672" /></p>
<div class="sourceCode" id="cb179"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb179-1"><a href="visium-mouse-coronal-workflow.html#cb179-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb179-2"><a href="visium-mouse-coronal-workflow.html#cb179-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-3"><a href="visium-mouse-coronal-workflow.html#cb179-3" aria-hidden="true" tabindex="-1"></a><span class="co"># select QC thresholds</span></span>
<span id="cb179-4"><a href="visium-mouse-coronal-workflow.html#cb179-4" aria-hidden="true" tabindex="-1"></a>qc_lib_size <span class="ot">&lt;-</span> <span class="fu">colData</span>(spe)<span class="sc">$</span>sum <span class="sc">&lt;</span> <span class="dv">5000</span></span>
<span id="cb179-5"><a href="visium-mouse-coronal-workflow.html#cb179-5" aria-hidden="true" tabindex="-1"></a>qc_detected <span class="ot">&lt;-</span> <span class="fu">colData</span>(spe)<span class="sc">$</span>detected <span class="sc">&lt;</span> <span class="dv">1000</span></span>
<span id="cb179-6"><a href="visium-mouse-coronal-workflow.html#cb179-6" aria-hidden="true" tabindex="-1"></a>qc_mito <span class="ot">&lt;-</span> <span class="fu">colData</span>(spe)<span class="sc">$</span>subsets_mito_percent <span class="sc">&gt;</span> <span class="dv">30</span></span>
<span id="cb179-7"><a href="visium-mouse-coronal-workflow.html#cb179-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-8"><a href="visium-mouse-coronal-workflow.html#cb179-8" aria-hidden="true" tabindex="-1"></a><span class="co"># number of discarded spots for each QC metric</span></span>
<span id="cb179-9"><a href="visium-mouse-coronal-workflow.html#cb179-9" aria-hidden="true" tabindex="-1"></a><span class="fu">apply</span>(<span class="fu">cbind</span>(qc_lib_size, qc_detected, qc_mito), <span class="dv">2</span>, sum)</span></code></pre></div>
<pre><code>## qc_lib_size qc_detected     qc_mito 
##           9           4          11</code></pre>
<div class="sourceCode" id="cb181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb181-1"><a href="visium-mouse-coronal-workflow.html#cb181-1" aria-hidden="true" tabindex="-1"></a><span class="co"># combined set of discarded spots</span></span>
<span id="cb181-2"><a href="visium-mouse-coronal-workflow.html#cb181-2" aria-hidden="true" tabindex="-1"></a>discard <span class="ot">&lt;-</span> qc_lib_size <span class="sc">|</span> qc_detected <span class="sc">|</span> qc_mito</span>
<span id="cb181-3"><a href="visium-mouse-coronal-workflow.html#cb181-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(discard)</span></code></pre></div>
<pre><code>## discard
## FALSE  TRUE 
##  2683    19</code></pre>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="visium-mouse-coronal-workflow.html#cb183-1" aria-hidden="true" tabindex="-1"></a><span class="co"># store in object</span></span>
<span id="cb183-2"><a href="visium-mouse-coronal-workflow.html#cb183-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(spe)<span class="sc">$</span>discard <span class="ot">&lt;-</span> discard</span></code></pre></div>
<p>Plot discarded spots in x-y coordinates on the tissue slide to check if there is any biologically meaningful spatial pattern. This would be problematic, since it would mean we are removing biologically informative spots.</p>
<p>We use QC visualization functions from the <a href="https://github.com/lmweber/spatzli">spatzli</a> package to generate plots.</p>
<div class="sourceCode" id="cb184"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb184-1"><a href="visium-mouse-coronal-workflow.html#cb184-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(spatzli)</span>
<span id="cb184-2"><a href="visium-mouse-coronal-workflow.html#cb184-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-3"><a href="visium-mouse-coronal-workflow.html#cb184-3" aria-hidden="true" tabindex="-1"></a><span class="co"># check spatial pattern of discarded spots</span></span>
<span id="cb184-4"><a href="visium-mouse-coronal-workflow.html#cb184-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQCspots</span>(spe, <span class="at">discard =</span> <span class="st">&quot;discard&quot;</span>)</span></code></pre></div>
<p><img src="workflow_Visium_mouseCoronal_files/figure-html/QC_check-1.png" width="672" /></p>
<p>There is one small region with some concentrated discarded spots at the top-left. However, this does not appear to correspond to any specific known anatomical region of interest. We assume that these are low-quality spots, and filtering them out will not cause problems in the biological interpretation.</p>
<p>We filter out the low-quality spots from the object.</p>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="visium-mouse-coronal-workflow.html#cb185-1" aria-hidden="true" tabindex="-1"></a><span class="co"># filter low-quality spots</span></span>
<span id="cb185-2"><a href="visium-mouse-coronal-workflow.html#cb185-2" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> spe[, <span class="sc">!</span><span class="fu">colData</span>(spe)<span class="sc">$</span>discard]</span>
<span id="cb185-3"><a href="visium-mouse-coronal-workflow.html#cb185-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(spe)</span></code></pre></div>
<pre><code>## [1] 32285  2683</code></pre>
</div>
<div id="normalization-2" class="section level2" number="18.5">
<h2><span class="header-section-number">18.5</span> Normalization</h2>
<p>Next, we calculate log-transformed normalized counts, using pool-based size factors and deconvolution to the spot level, using methods from <code>scater</code> <span class="citation">(<a href="references.html#ref-McCarthy2017" role="doc-biblioref">McCarthy et al. 2017</a>)</span> and <code>scran</code> <span class="citation">(<a href="references.html#ref-Lun2016" role="doc-biblioref">Lun, McCarthy, and Marioni 2016</a>)</span>. Since we have a single sample, there are no blocking factors in the experimental design.</p>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="visium-mouse-coronal-workflow.html#cb187-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scran)</span>
<span id="cb187-2"><a href="visium-mouse-coronal-workflow.html#cb187-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-3"><a href="visium-mouse-coronal-workflow.html#cb187-3" aria-hidden="true" tabindex="-1"></a><span class="co"># quick clustering for pool-based size factors</span></span>
<span id="cb187-4"><a href="visium-mouse-coronal-workflow.html#cb187-4" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb187-5"><a href="visium-mouse-coronal-workflow.html#cb187-5" aria-hidden="true" tabindex="-1"></a>qclus <span class="ot">&lt;-</span> <span class="fu">quickCluster</span>(spe)</span>
<span id="cb187-6"><a href="visium-mouse-coronal-workflow.html#cb187-6" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(qclus)</span></code></pre></div>
<pre><code>## qclus
##   1   2   3   4   5   6   7   8   9  10  11  12 
## 178 162 255 450 274 235 320 250 105 119 131 204</code></pre>
<div class="sourceCode" id="cb189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb189-1"><a href="visium-mouse-coronal-workflow.html#cb189-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate size factors and store in object</span></span>
<span id="cb189-2"><a href="visium-mouse-coronal-workflow.html#cb189-2" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> <span class="fu">computeSumFactors</span>(spe, <span class="at">cluster =</span> qclus)</span>
<span id="cb189-3"><a href="visium-mouse-coronal-workflow.html#cb189-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4"><a href="visium-mouse-coronal-workflow.html#cb189-4" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">sizeFactors</span>(spe))</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.1009  0.6032  0.8985  1.0000  1.2649  6.0391</code></pre>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="visium-mouse-coronal-workflow.html#cb191-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">log10</span>(<span class="fu">sizeFactors</span>(spe)), <span class="at">xlab =</span> <span class="st">&quot;log10 (size factors)&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Size factors&quot;</span>)</span></code></pre></div>
<p><img src="workflow_Visium_mouseCoronal_files/figure-html/normalization-1.png" width="672" /></p>
<div class="sourceCode" id="cb192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb192-1"><a href="visium-mouse-coronal-workflow.html#cb192-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate logcounts (log-transformed normalized counts) and store in object</span></span>
<span id="cb192-2"><a href="visium-mouse-coronal-workflow.html#cb192-2" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> <span class="fu">logNormCounts</span>(spe)</span>
<span id="cb192-3"><a href="visium-mouse-coronal-workflow.html#cb192-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-4"><a href="visium-mouse-coronal-workflow.html#cb192-4" aria-hidden="true" tabindex="-1"></a><span class="fu">assayNames</span>(spe)</span></code></pre></div>
<pre><code>## [1] &quot;counts&quot;    &quot;logcounts&quot;</code></pre>
</div>
<div id="feature-selection-2" class="section level2" number="18.6">
<h2><span class="header-section-number">18.6</span> Feature selection</h2>
<p>Identify a set of top highly variable genes (HVGs), which will be used to define cell types. We use methods from <code>scran</code> <span class="citation">(<a href="references.html#ref-Lun2016" role="doc-biblioref">Lun, McCarthy, and Marioni 2016</a>)</span>, and first filter out mitochondrial genes (since these are very highly expressed and not of biological interest here).</p>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="visium-mouse-coronal-workflow.html#cb194-1" aria-hidden="true" tabindex="-1"></a><span class="co"># remove mitochondrial genes</span></span>
<span id="cb194-2"><a href="visium-mouse-coronal-workflow.html#cb194-2" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> spe[<span class="sc">!</span>is_mito, ]</span>
<span id="cb194-3"><a href="visium-mouse-coronal-workflow.html#cb194-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(spe)</span></code></pre></div>
<pre><code>## [1] 32272  2683</code></pre>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="visium-mouse-coronal-workflow.html#cb196-1" aria-hidden="true" tabindex="-1"></a><span class="co"># fit mean-variance relationship</span></span>
<span id="cb196-2"><a href="visium-mouse-coronal-workflow.html#cb196-2" aria-hidden="true" tabindex="-1"></a>dec <span class="ot">&lt;-</span> <span class="fu">modelGeneVar</span>(spe)</span>
<span id="cb196-3"><a href="visium-mouse-coronal-workflow.html#cb196-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb196-4"><a href="visium-mouse-coronal-workflow.html#cb196-4" aria-hidden="true" tabindex="-1"></a><span class="co"># visualize mean-variance relationship</span></span>
<span id="cb196-5"><a href="visium-mouse-coronal-workflow.html#cb196-5" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">metadata</span>(dec)</span>
<span id="cb196-6"><a href="visium-mouse-coronal-workflow.html#cb196-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(fit<span class="sc">$</span>mean, fit<span class="sc">$</span>var, </span>
<span id="cb196-7"><a href="visium-mouse-coronal-workflow.html#cb196-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;mean of log-expression&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;variance of log-expression&quot;</span>)</span>
<span id="cb196-8"><a href="visium-mouse-coronal-workflow.html#cb196-8" aria-hidden="true" tabindex="-1"></a><span class="fu">curve</span>(fit<span class="sc">$</span><span class="fu">trend</span>(x), <span class="at">col =</span> <span class="st">&quot;dodgerblue&quot;</span>, <span class="at">add =</span> <span class="cn">TRUE</span>, <span class="at">lwd =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="workflow_Visium_mouseCoronal_files/figure-html/feature_selection_HVGs-1.png" width="672" /></p>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="visium-mouse-coronal-workflow.html#cb197-1" aria-hidden="true" tabindex="-1"></a><span class="co"># select top HVGs</span></span>
<span id="cb197-2"><a href="visium-mouse-coronal-workflow.html#cb197-2" aria-hidden="true" tabindex="-1"></a>top_hvgs <span class="ot">&lt;-</span> <span class="fu">getTopHVGs</span>(dec, <span class="at">prop =</span> <span class="fl">0.1</span>)</span>
<span id="cb197-3"><a href="visium-mouse-coronal-workflow.html#cb197-3" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(top_hvgs)</span></code></pre></div>
<pre><code>## [1] 1216</code></pre>
<p>Note there are a few extremely highly expressed genes, which influence the fitted mean-variance relationship. We check the names of these genes to decide whether they should be removed as outliers.</p>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="visium-mouse-coronal-workflow.html#cb199-1" aria-hidden="true" tabindex="-1"></a><span class="co"># identify outlier genes</span></span>
<span id="cb199-2"><a href="visium-mouse-coronal-workflow.html#cb199-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rev</span>(<span class="fu">sort</span>(fit<span class="sc">$</span>mean))[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>## ENSMUSG00000115783 ENSMUSG00000098178 ENSMUSG00000024661 
##          10.992042           9.904624           7.370274</code></pre>
<div class="sourceCode" id="cb201"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb201-1"><a href="visium-mouse-coronal-workflow.html#cb201-1" aria-hidden="true" tabindex="-1"></a>outlier_ids <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">rev</span>(<span class="fu">sort</span>(fit<span class="sc">$</span>mean))[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>])</span>
<span id="cb201-2"><a href="visium-mouse-coronal-workflow.html#cb201-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb201-3"><a href="visium-mouse-coronal-workflow.html#cb201-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(spe)[outlier_ids, ]</span></code></pre></div>
<pre><code>## DataFrame with 3 rows and 3 columns
##                               gene_id   gene_name    feature_type
##                           &lt;character&gt; &lt;character&gt;     &lt;character&gt;
## ENSMUSG00000115783 ENSMUSG00000115783         Bc1 Gene Expression
## ENSMUSG00000098178 ENSMUSG00000098178     Gm42418 Gene Expression
## ENSMUSG00000024661 ENSMUSG00000024661        Fth1 Gene Expression</code></pre>
<p>These appear to be biologically meaningful genes, so we leave them in.</p>
</div>
<div id="dimensionality-reduction-2" class="section level2" number="18.7">
<h2><span class="header-section-number">18.7</span> Dimensionality reduction</h2>
<p>Run principal component analysis (PCA) using <code>scater</code> <span class="citation">(<a href="references.html#ref-McCarthy2017" role="doc-biblioref">McCarthy et al. 2017</a>)</span>, and retain the top 50 principal components (PCs) for downstream analyses. Also run UMAP on the top 50 PCs, and retain the top 2 UMAP components for visualization purposes.</p>
<div class="sourceCode" id="cb203"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb203-1"><a href="visium-mouse-coronal-workflow.html#cb203-1" aria-hidden="true" tabindex="-1"></a><span class="co"># compute PCA</span></span>
<span id="cb203-2"><a href="visium-mouse-coronal-workflow.html#cb203-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb203-3"><a href="visium-mouse-coronal-workflow.html#cb203-3" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(spe, <span class="at">subset_row =</span> top_hvgs)</span>
<span id="cb203-4"><a href="visium-mouse-coronal-workflow.html#cb203-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb203-5"><a href="visium-mouse-coronal-workflow.html#cb203-5" aria-hidden="true" tabindex="-1"></a><span class="fu">reducedDimNames</span>(spe)</span></code></pre></div>
<pre><code>## [1] &quot;PCA&quot;</code></pre>
<div class="sourceCode" id="cb205"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb205-1"><a href="visium-mouse-coronal-workflow.html#cb205-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">reducedDim</span>(spe, <span class="st">&quot;PCA&quot;</span>))</span></code></pre></div>
<pre><code>## [1] 2683   50</code></pre>
<div class="sourceCode" id="cb207"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb207-1"><a href="visium-mouse-coronal-workflow.html#cb207-1" aria-hidden="true" tabindex="-1"></a><span class="co"># compute UMAP on top 50 PCs</span></span>
<span id="cb207-2"><a href="visium-mouse-coronal-workflow.html#cb207-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb207-3"><a href="visium-mouse-coronal-workflow.html#cb207-3" aria-hidden="true" tabindex="-1"></a>spe <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(spe, <span class="at">dimred =</span> <span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb207-4"><a href="visium-mouse-coronal-workflow.html#cb207-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb207-5"><a href="visium-mouse-coronal-workflow.html#cb207-5" aria-hidden="true" tabindex="-1"></a><span class="fu">reducedDimNames</span>(spe)</span></code></pre></div>
<pre><code>## [1] &quot;PCA&quot;  &quot;UMAP&quot;</code></pre>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="visium-mouse-coronal-workflow.html#cb209-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">reducedDim</span>(spe, <span class="st">&quot;UMAP&quot;</span>))</span></code></pre></div>
<pre><code>## [1] 2683    2</code></pre>
<div class="sourceCode" id="cb211"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb211-1"><a href="visium-mouse-coronal-workflow.html#cb211-1" aria-hidden="true" tabindex="-1"></a><span class="co"># update column names for plotting functions</span></span>
<span id="cb211-2"><a href="visium-mouse-coronal-workflow.html#cb211-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(<span class="fu">reducedDim</span>(spe, <span class="st">&quot;UMAP&quot;</span>)) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;UMAP&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="clustering-2" class="section level2" number="18.8">
<h2><span class="header-section-number">18.8</span> Clustering</h2>
<p>Perform clustering to define cell types. We apply graph-based clustering using the Walktrap method implemented in <code>scran</code> <span class="citation">(<a href="references.html#ref-Lun2016" role="doc-biblioref">Lun, McCarthy, and Marioni 2016</a>)</span>, applied to the top 50 PCs calculated on the set of top HVGs.</p>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="visium-mouse-coronal-workflow.html#cb212-1" aria-hidden="true" tabindex="-1"></a><span class="co"># graph-based clustering</span></span>
<span id="cb212-2"><a href="visium-mouse-coronal-workflow.html#cb212-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb212-3"><a href="visium-mouse-coronal-workflow.html#cb212-3" aria-hidden="true" tabindex="-1"></a>k <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb212-4"><a href="visium-mouse-coronal-workflow.html#cb212-4" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">buildSNNGraph</span>(spe, <span class="at">k =</span> k, <span class="at">use.dimred =</span> <span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb212-5"><a href="visium-mouse-coronal-workflow.html#cb212-5" aria-hidden="true" tabindex="-1"></a>g_walk <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">cluster_walktrap</span>(g)</span>
<span id="cb212-6"><a href="visium-mouse-coronal-workflow.html#cb212-6" aria-hidden="true" tabindex="-1"></a>clus <span class="ot">&lt;-</span> g_walk<span class="sc">$</span>membership</span>
<span id="cb212-7"><a href="visium-mouse-coronal-workflow.html#cb212-7" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(clus)</span></code></pre></div>
<pre><code>## clus
##   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20 
## 125 126  61 229 241 283 165 132 161 267 122  43  99  94 101 161  26  58  64  24 
##  21  22  23 
##  51  32  18</code></pre>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="visium-mouse-coronal-workflow.html#cb214-1" aria-hidden="true" tabindex="-1"></a><span class="co"># store cluster labels in column &#39;label&#39; in colData</span></span>
<span id="cb214-2"><a href="visium-mouse-coronal-workflow.html#cb214-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colLabels</span>(spe) <span class="ot">&lt;-</span> <span class="fu">factor</span>(clus)</span></code></pre></div>
<p>Visualize the clusters by plotting in (i) spatial x-y coordinates on the tissue slide, and (ii) UMAP dimensions. We use plotting functions from the <a href="https://github.com/lmweber/ggpavis">ggspavis</a> package.</p>
<div class="sourceCode" id="cb215"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb215-1"><a href="visium-mouse-coronal-workflow.html#cb215-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggspavis)</span>
<span id="cb215-2"><a href="visium-mouse-coronal-workflow.html#cb215-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb215-3"><a href="visium-mouse-coronal-workflow.html#cb215-3" aria-hidden="true" tabindex="-1"></a><span class="co"># define custom color palette</span></span>
<span id="cb215-4"><a href="visium-mouse-coronal-workflow.html#cb215-4" aria-hidden="true" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">palette.colors</span>(<span class="at">palette =</span> <span class="st">&quot;Polychrome 36&quot;</span>))</span>
<span id="cb215-5"><a href="visium-mouse-coronal-workflow.html#cb215-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb215-6"><a href="visium-mouse-coronal-workflow.html#cb215-6" aria-hidden="true" tabindex="-1"></a><span class="co"># plot clusters in spatial x-y coordinates</span></span>
<span id="cb215-7"><a href="visium-mouse-coronal-workflow.html#cb215-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSpots</span>(spe, <span class="at">discrete =</span> <span class="st">&quot;label&quot;</span>, <span class="at">palette =</span> colors)</span></code></pre></div>
<p><img src="workflow_Visium_mouseCoronal_files/figure-html/clustering_plots_spots-1.png" width="672" /></p>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="visium-mouse-coronal-workflow.html#cb216-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot clusters in UMAP dimensions</span></span>
<span id="cb216-2"><a href="visium-mouse-coronal-workflow.html#cb216-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDimRed</span>(spe, <span class="at">type =</span> <span class="st">&quot;UMAP&quot;</span>, <span class="at">discrete =</span> <span class="st">&quot;label&quot;</span>, <span class="at">palette =</span> colors)</span></code></pre></div>
<p><img src="workflow_Visium_mouseCoronal_files/figure-html/clustering_plots_reduced-1.png" width="480" /></p>
</div>
<div id="marker-genes-3" class="section level2" number="18.9">
<h2><span class="header-section-number">18.9</span> Marker genes</h2>
<p>Identify marker genes by testing for differential gene expression between clusters, using the binomial test implemented in <code>findMarkers</code> in <code>scran</code> <span class="citation">(<a href="references.html#ref-Lun2016" role="doc-biblioref">Lun, McCarthy, and Marioni 2016</a>)</span>.</p>
<div class="sourceCode" id="cb217"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb217-1"><a href="visium-mouse-coronal-workflow.html#cb217-1" aria-hidden="true" tabindex="-1"></a><span class="co"># set gene names as row names for visualization purposes</span></span>
<span id="cb217-2"><a href="visium-mouse-coronal-workflow.html#cb217-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(spe) <span class="ot">&lt;-</span> <span class="fu">rowData</span>(spe)<span class="sc">$</span>gene_name</span>
<span id="cb217-3"><a href="visium-mouse-coronal-workflow.html#cb217-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb217-4"><a href="visium-mouse-coronal-workflow.html#cb217-4" aria-hidden="true" tabindex="-1"></a><span class="co"># test for marker genes</span></span>
<span id="cb217-5"><a href="visium-mouse-coronal-workflow.html#cb217-5" aria-hidden="true" tabindex="-1"></a>markers <span class="ot">&lt;-</span> <span class="fu">findMarkers</span>(spe, <span class="at">test =</span> <span class="st">&quot;binom&quot;</span>, <span class="at">direction =</span> <span class="st">&quot;up&quot;</span>)</span>
<span id="cb217-6"><a href="visium-mouse-coronal-workflow.html#cb217-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb217-7"><a href="visium-mouse-coronal-workflow.html#cb217-7" aria-hidden="true" tabindex="-1"></a><span class="co"># returns a list with one DataFrame per cluster</span></span>
<span id="cb217-8"><a href="visium-mouse-coronal-workflow.html#cb217-8" aria-hidden="true" tabindex="-1"></a>markers</span></code></pre></div>
<pre><code>## List of length 23
## names(23): 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23</code></pre>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb219-1"><a href="visium-mouse-coronal-workflow.html#cb219-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb219-2"><a href="visium-mouse-coronal-workflow.html#cb219-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb219-3"><a href="visium-mouse-coronal-workflow.html#cb219-3" aria-hidden="true" tabindex="-1"></a><span class="co"># plot log-fold changes for one cluster over all other clusters</span></span>
<span id="cb219-4"><a href="visium-mouse-coronal-workflow.html#cb219-4" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting cluster 5</span></span>
<span id="cb219-5"><a href="visium-mouse-coronal-workflow.html#cb219-5" aria-hidden="true" tabindex="-1"></a>interesting <span class="ot">&lt;-</span> markers[[<span class="dv">5</span>]]</span>
<span id="cb219-6"><a href="visium-mouse-coronal-workflow.html#cb219-6" aria-hidden="true" tabindex="-1"></a>best_set <span class="ot">&lt;-</span> interesting[interesting<span class="sc">$</span>Top <span class="sc">&lt;=</span> <span class="dv">5</span>, ]</span>
<span id="cb219-7"><a href="visium-mouse-coronal-workflow.html#cb219-7" aria-hidden="true" tabindex="-1"></a>logFCs <span class="ot">&lt;-</span> <span class="fu">getMarkerEffects</span>(best_set)</span>
<span id="cb219-8"><a href="visium-mouse-coronal-workflow.html#cb219-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb219-9"><a href="visium-mouse-coronal-workflow.html#cb219-9" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(logFCs, <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">5</span>, <span class="at">length.out =</span> <span class="dv">101</span>))</span></code></pre></div>
<p><img src="workflow_Visium_mouseCoronal_files/figure-html/marker_genes_heatmap-1.png" width="672" /></p>
<div class="sourceCode" id="cb220"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb220-1"><a href="visium-mouse-coronal-workflow.html#cb220-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot log-transformed normalized expression of top genes for one cluster</span></span>
<span id="cb220-2"><a href="visium-mouse-coronal-workflow.html#cb220-2" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> <span class="fu">head</span>(<span class="fu">rownames</span>(interesting))</span>
<span id="cb220-3"><a href="visium-mouse-coronal-workflow.html#cb220-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb220-4"><a href="visium-mouse-coronal-workflow.html#cb220-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotExpression</span>(spe, <span class="at">x =</span> <span class="st">&quot;label&quot;</span>, <span class="at">features =</span> top_genes)</span></code></pre></div>
<p><img src="workflow_Visium_mouseCoronal_files/figure-html/marker_genes_expression-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="visium-human-dlpfc-workflow.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visium-spatiallibd-workflow.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OSTA-book.pdf", "OSTA-book.epub"],
"toc": {
"collapse": "section"
},
"lib_dir": "book_assets"
});
});
</script>

</body>

</html>
